<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="On-target gene knockdown using siRNA ideally results from binding 
    fully complementary regions in mRNA transcripts to induce cleavage. 
    Off-target siRNA gene knockdown can occur through several modes, one being 
    a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated 
    off-target effects occur when the ~8 nucleotides at the 5’ end of the guide 
    strand, called a seed region, bind the 3’ untranslated regions of mRNA, 
    causing reduced translation. Experiments using siRNA knockdown paired with 
    RNA-seq can be used to detect siRNA sequences with potential off-target 
    effects driven by the seed region. SeedMatchR provides tools for exploring 
    and detecting potential seed-mediated off-target effects of siRNA in RNA-seq 
    experiments. SeedMatchR is designed to extend current differential 
    expression analysis tools, such as DESeq2, by annotating results with 
    predicted seed matches. Using publicly available data, we demonstrate the 
    ability of SeedMatchR to detect cumulative changes in differential gene 
    expression attributed to siRNA seed regions. ">
<title>Find Matches to Canonical SiRNA Seeds in Genomic Features • SeedMatchR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Find Matches to Canonical SiRNA Seeds in Genomic Features">
<meta property="og:description" content="On-target gene knockdown using siRNA ideally results from binding 
    fully complementary regions in mRNA transcripts to induce cleavage. 
    Off-target siRNA gene knockdown can occur through several modes, one being 
    a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated 
    off-target effects occur when the ~8 nucleotides at the 5’ end of the guide 
    strand, called a seed region, bind the 3’ untranslated regions of mRNA, 
    causing reduced translation. Experiments using siRNA knockdown paired with 
    RNA-seq can be used to detect siRNA sequences with potential off-target 
    effects driven by the seed region. SeedMatchR provides tools for exploring 
    and detecting potential seed-mediated off-target effects of siRNA in RNA-seq 
    experiments. SeedMatchR is designed to extend current differential 
    expression analysis tools, such as DESeq2, by annotating results with 
    predicted seed matches. Using publicly available data, we demonstrate the 
    ability of SeedMatchR to detect cumulative changes in differential gene 
    expression attributed to siRNA seed regions. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/SeedMatchR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="seedmatchr">SeedMatchR<a class="anchor" aria-label="anchor" href="#seedmatchr"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of SeedMatchR is to help users identify potential seed-mediated effects in their RNAseq data.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of SeedMatchR from <a href="https://github.com/" class="external-link">GitHub</a> or the stable build from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tacazares/SeedMatchR"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="troubleshooting-installation-issues">Troubleshooting installation issues<a class="anchor" aria-label="anchor" href="#troubleshooting-installation-issues"></a>
</h3>
<p>There have been some issues with installing <code>SeedMatchR</code> on Mac OS and Linux, especially if you use the most recent version of R (≥ 4.3). One potential suggestion is to use the version of R that was used in development (4.1.2), though <code>SeedMatchR</code> should work on the most recent version as well.</p>
<p>In addition, some users have issues with <code>ggmsa</code>. The package <code>ggmsa</code> requires the installations of the <code>msa</code> package to work. There have been issues with using the <code><a href="reference/plot_seeds.html">plot_seeds()</a></code> function that cause errors due to a missing loaded package. To fix this, users must install and load the <code>msa</code> package from CRAN, in addition to <code>SeedMatchR</code>.</p>
<p><code>SeedMatchR</code> has been installed and tested on the most recent builds of R for Mac OS, Windows, and Linux. The results of the build tests are found here: <a href="https://github.com/tacazares/SeedMatchR/actions/runs/6634297920" class="external-link uri">https://github.com/tacazares/SeedMatchR/actions/runs/6634297920</a></p>
<p>The file containing the working versions of each dependency is listed in the GitHub actions logs. For example, the most recent Mac OS build dependencies that work can be found in this <a href="https://github.com/tacazares/SeedMatchR/actions/runs/6634297920/job/18023503364#step:5:5857" class="external-link">log</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="quick-start-example-with-public-sirna-data">Quick start example with public siRNA data<a class="anchor" aria-label="anchor" href="#quick-start-example-with-public-sirna-data"></a>
</h2>
<p>This example uses the siRNA sequence, D1, targeting the Ttr gene in rat liver from the publication:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga<span class="sc">-</span>Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp<span class="sc">-</span>Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside<span class="sc">:</span> Improving the clinical safety of GalNAc<span class="sc">-</span>siRNA conjugates using seed<span class="sc">-</span>pairing destabilization. Nucleic Acids Res. <span class="dv">2022</span> Jul <span class="dv">8</span>;<span class="dv">50</span>(<span class="dv">12</span>)<span class="sc">:</span><span class="dv">6656</span><span class="fl">-6670.</span> doi<span class="sc">:</span> <span class="fl">10.1093</span><span class="sc">/</span>nar<span class="sc">/</span>gkac539. PMID<span class="sc">:</span> <span class="dv">35736224</span>; PMCID<span class="sc">:</span> PMC9262600.</span></code></pre></div>
<p>The guide sequence of interest is 23 bp long and oriented 5’ -&gt; 3’.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># siRNA sequence of interest targeting a 23 bp region of the Ttr gene</span></span>
<span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-rat-specific-annotation-data">Load rat specific annotation data.<a class="anchor" aria-label="anchor" href="#load-rat-specific-annotation-data"></a>
</h3>
<p>We use <code>AnnotationHub</code> to derive the <code>GTF</code> and <code>DNA</code> sequence files for the species of interest. Once you have derived the annotations, you could save them as an Rdata object to increase the speed of loading the datasets.</p>
<div class="section level4">
<h4 id="load-annotation-databases">Load annotation databases<a class="anchor" aria-label="anchor" href="#load-annotation-databases"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the species specific annotation database object</span></span>
<span><span class="va">anno.db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/load_species_anno_db.html">load_species_anno_db</a></span><span class="op">(</span><span class="st">"rat"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="extract-features-and-sequences-of-interest-from-annotations">Extract features and sequences of interest from annotations<a class="anchor" aria-label="anchor" href="#extract-features-and-sequences-of-interest-from-annotations"></a>
</h4>
<p>We will use the annotations to derive the features and feature sequences that we want to scan for each gene.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">features</span> <span class="op">=</span> <span class="fu"><a href="reference/get_feature_seqs.html">get_feature_seqs</a></span><span class="op">(</span><span class="va">anno.db</span><span class="op">$</span><span class="va">tx.db</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">dna</span>, feature.type <span class="op">=</span> <span class="st">"3UTR"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="prepare-deseq2-results">Prepare DESEQ2 Results<a class="anchor" aria-label="anchor" href="#prepare-deseq2-results"></a>
</h3>
<p>The test data that is provided with <code>SeedMatchR</code> was derived from the 2022 publication by Schlegel et al. The data set represents a DESeq2 analysis performed on rat liver that had been treated with Ttr targeting siRNA. We will use this example to explore seed mediated activity.</p>
<div class="section level4">
<h4 id="download-data-only-need-to-perform-once-can-skip-to-loading-if-done">Download data (only need to perform once, can skip to loading if done)<a class="anchor" aria-label="anchor" href="#download-data-only-need-to-perform-once-can-skip-to-loading-if-done"></a>
</h4>
<p>We start by downloading the example data set. This function will download three files from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929" class="external-link">GSE184929</a>. These files represent three samples with different siRNA treatments at two dosages.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h4>
<p>We can load the example data into the environment.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sirna.data</span> <span class="op">=</span> <span class="fu"><a href="reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
<p>The DESeq2 results are available through the names <code>Schlegel_2022_Ttr_D1_30mkg</code>, <code>Schlegel_2022_Ttr_D4_30mkg</code> and <code>Schlegel_2022_Ttr_D1_10mkg</code>. The data set name is long, so it will be renamed to <code>res</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">sirna.data</span><span class="op">$</span><span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filter-example-results">Filter example results<a class="anchor" aria-label="anchor" href="#filter-example-results"></a>
</h4>
<p>The DESeq2 results file is then filtered. The function <code><a href="reference/filter_deseq.html">filter_deseq()</a></code> can be used to filter a results file by log2FoldChange, padj, baseMean, and remove NA entries.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dimensions before filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 32883    6</span></span>
<span><span class="co">#&gt; [1] 32883     8</span></span>
<span></span>
<span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="reference/filter_deseq.html">filter_deseq</a></span><span class="op">(</span><span class="va">res</span>, fdr.cutoff<span class="op">=</span><span class="fl">1</span>, fc.cutoff<span class="op">=</span><span class="fl">0</span>, rm.na.log2fc <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Dimensions after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 13582     8</span></span>
<span><span class="co">#&gt; [1] 13582     8</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="counting-seed-matches-in-transcripts">Counting seed matches in transcripts<a class="anchor" aria-label="anchor" href="#counting-seed-matches-in-transcripts"></a>
</h3>
<p>You can perform a seed match for a single seed using the <code><a href="reference/SeedMatchR.html">SeedMatchR()</a></code> function.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, <span class="va">guide.seq</span>, <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0</span></span></code></pre></div>
<div class="section level4">
<h4 id="match-multiple-seeds">Match multiple seeds<a class="anchor" aria-label="anchor" href="#match-multiple-seeds"></a>
</h4>
<p>You can perform seed matching for all available seeds using a for loop. The results will be appended as a new column to the results data frame.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">seed</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mer8"</span>, <span class="st">"mer6"</span>, <span class="st">"mer7A1"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, <span class="va">guide.seq</span>, seed.name <span class="op">=</span> <span class="va">seed</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8 mer8 mer6 mer7A1</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1    1    1      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0    0    0      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1    0    1      0</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0    0    0      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0    0    0      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0    0    0      0</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="comparing-the-expression-profiles-of-seed-targets-to-background">Comparing the expression profiles of seed targets to background<a class="anchor" aria-label="anchor" href="#comparing-the-expression-profiles-of-seed-targets-to-background"></a>
</h3>
<p>Many factors that perturb gene expression, like miRNA, show cumulative changes in their targets gene expression. Cumulative changes in the profile of genes expression can be visualized and tested with the emperical distribution function (ecdf) coupled with a statistical test such as the Kolmogorov-Smirnov test.</p>
<p><code>SeedMatchR</code> provides functions for comparing the log2(Fold Change) of two gene sets. The function <code>deseq_fc_ecdf</code> is designed to work directly with a DESeq2 results data frame.</p>
<p>Required Inputs:</p>
<ul>
<li>
<code>res</code>: DESeq2 results data frame</li>
<li>
<code>gene.lists</code>: A list of lists containing gene names</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene set 1 </span></span>
<span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Gene set 2</span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="reference/de_fc_ecdf.html">de_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, </span>
<span>                                  <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span><span class="op">)</span>,</span>
<span>                             stats.test <span class="op">=</span> <span class="st">"KS"</span>, </span>
<span>                             factor.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span>, </span>
<span>                                              <span class="st">"mer7m8"</span><span class="op">)</span>, </span>
<span>                             null.name <span class="op">=</span> <span class="st">"Background"</span>,</span>
<span>                             target.name <span class="op">=</span> <span class="st">"mer7m8"</span>,<span class="op">)</span></span>
<span><span class="co">#&gt; Comparing: Background vs. mer7m8</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-13-1.png" width="100%"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=SeedMatchR" class="external-link">View on CRAN</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SeedMatchR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Tareian Cazares <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-4624-6156" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Gulcin Ozer <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8315-8051" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jibo Wang <br><small class="roles"> Author </small>  </li>
<li>Rick Higgs <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/tacazares/SeedMatchR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/tacazares/SeedMatchR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tareian Cazares, Gulcin Ozer, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
