<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find siRNA Matches in Genomic Features • SeedMatchR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Find siRNA Matches in Genomic Features">
<meta name="description" content="On-target siRNA silencing of mRNA results from binding to fully complementary regions in transcripts to induce cleavage. Off-target siRNA gene knockdown can occur through several modes, one being a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated off-target effects occur when the ~8 nucleotides at the 5’ end of the guide strand, called a seed region, bind the 3’ untranslated regions of mRNA, causing reduced translation. Experiments using siRNA knockdown paired with RNA-seq can be used to detect siRNA sequences with potential off-target effects driven by the seed region. SeedMatchR provides tools for exploring and detecting potential seed-mediated off-target effects of siRNA in RNA-seq experiments. SeedMatchR is designed to extend current differential expression analysis tools, such as DESeq2, by annotating results with predicted seed matches. Using publicly available data, we demonstrate the ability of SeedMatchR to detect cumulative changes in differential gene expression attributed to siRNA seed regions.">
<meta property="og:description" content="On-target siRNA silencing of mRNA results from binding to fully complementary regions in transcripts to induce cleavage. Off-target siRNA gene knockdown can occur through several modes, one being a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated off-target effects occur when the ~8 nucleotides at the 5’ end of the guide strand, called a seed region, bind the 3’ untranslated regions of mRNA, causing reduced translation. Experiments using siRNA knockdown paired with RNA-seq can be used to detect siRNA sequences with potential off-target effects driven by the seed region. SeedMatchR provides tools for exploring and detecting potential seed-mediated off-target effects of siRNA in RNA-seq experiments. SeedMatchR is designed to extend current differential expression analysis tools, such as DESeq2, by annotating results with predicted seed matches. Using publicly available data, we demonstrate the ability of SeedMatchR to detect cumulative changes in differential gene expression attributed to siRNA seed regions.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/SeedMatchR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="seedmatchr-version-200">SeedMatchR version 2.0.0<a class="anchor" aria-label="anchor" href="#seedmatchr-version-200"></a>
</h1></div>
<p>The goal of SeedMatchR is to help users identify potential seed-mediated effects in their RNA-seq data.</p>
<p>These changes in this forked repository is to add the biological target bulges and wobbles in the search space.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>This version of SeedMatchR requires R ≥ 4.3.0, but it is recommended to use the latest version of R to avoid issues with annotation retrieval for newer genomes.</p>
<p>You can install the development version of SeedMatchR from <a href="https://github.com/" class="external-link">GitHub</a> or the stable build from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Public Repository</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tacazares/SeedMatchR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start-examples-with-public-sirna-data">Quick start examples with public siRNA data<a class="anchor" aria-label="anchor" href="#quick-start-examples-with-public-sirna-data"></a>
</h2>
<p>This example uses the siRNA sequence, D1, targeting the Ttr gene in rat liver from the publication:</p>
<blockquote>
<p>Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga-Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp-Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside: Improving the clinical safety of GalNAc-siRNA conjugates using seed-pairing destabilization. Nucleic Acids Res. 2022 Jul 8;50(12):6656-6670. doi: 10.1093/nar/gkac539. PMID: 35736224; PMCID: PMC9262600.</p>
</blockquote>
<p>The guide sequence of interest is 23 bp long and oriented 5’ -&gt; 3’.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># siRNA sequence of interest targeting a 23 bp region of the Ttr gene</span></span>
<span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-rat-specific-annotation-data">Load rat specific annotation data.<a class="anchor" aria-label="anchor" href="#load-rat-specific-annotation-data"></a>
</h3>
<p>We use <code>AnnotationHub</code> to derive the <code>GTF</code> and <code>DNA</code> sequence files for the species of interest. Once you have derived the annotations, you could save them as an Rdata object to increase the speed of loading the data sets. Running this function will take several minutes. Therefore it might be helpful to save the objects and reload them later if you plan to use this code in a repeated workflow.</p>
<div class="section level4">
<h4 id="load-annotation-databases">Load annotation databases<a class="anchor" aria-label="anchor" href="#load-annotation-databases"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annodb</span> <span class="op">=</span> <span class="fu"><a href="reference/load_annotations.html">load_annotations</a></span><span class="op">(</span>reference.name <span class="op">=</span> <span class="st">"rnor6"</span>, canonical <span class="op">=</span> <span class="cn">FALSE</span>, min.feature.width <span class="op">=</span> <span class="fl">8</span>, longest.utr <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; Build AnnotationFilter for transcript features based on the following parameters: </span></span>
<span><span class="co">#&gt; Keep only standard chroms: TRUE</span></span>
<span><span class="co">#&gt; Remove rows with NA in transcript ID: TRUE</span></span>
<span><span class="co">#&gt; Keep only protein coding genes and transcripts: TRUE</span></span>
<span><span class="co">#&gt; Filtering for transcripts with support level: FALSE</span></span>
<span><span class="co">#&gt; Keep only the ENSEMBL canonical transcript: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific genes: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific transcripts: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific gene symbols: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific entrez id: FALSE</span></span>
<span><span class="co">#&gt; Loading annotations from AnnotationHub for rnor6</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("rtracklayer")</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("ensembldb")</span></span>
<span><span class="co">#&gt; Extracting 3UTR from ensembldb object.</span></span>
<span><span class="co">#&gt; Keeping the longest UTR per gene.</span></span>
<span><span class="co">#&gt; Extracting sequences for each feature.</span></span>
<span><span class="co">#&gt; Keeping sequences that are &gt;= 8</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="example-1-perform-a-comprehensive-transcriptome-search">Example 1: Perform a comprehensive transcriptome search<a class="anchor" aria-label="anchor" href="#example-1-perform-a-comprehensive-transcriptome-search"></a>
</h2>
<p>The most straightforward way of using SeedMatchR is to search a reference set of transcripts given an input sequence.</p>
<div class="section level3">
<h3 id="output-match-ranges-as-granges">Output match ranges as granges<a class="anchor" aria-label="anchor" href="#output-match-ranges-as-granges"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.df</span> <span class="op">=</span> <span class="fu"><a href="reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                 sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                 seed.name <span class="op">=</span> <span class="st">"mer7m8"</span>, </span>
<span>                 res.format <span class="op">=</span> <span class="st">"granges"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="output-match-ranges-for-many-different-types-of-views-of-the-sirna">Output match ranges for many different types of views of the siRNA<a class="anchor" aria-label="anchor" href="#output-match-ranges-for-many-different-types-of-views-of-the-sirna"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.df</span> <span class="op">=</span> <span class="fu"><a href="reference/full_search.html">full_search</a></span><span class="op">(</span><span class="va">guide.seq</span>, <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, group.name <span class="op">=</span> <span class="st">"Ttr"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="example-2-analyze-rna-seq-data-with-seedmatchr">Example 2: Analyze RNA-seq data with SeedMatchR<a class="anchor" aria-label="anchor" href="#example-2-analyze-rna-seq-data-with-seedmatchr"></a>
</h2>
<div class="section level3">
<h3 id="prepare-deseq2-results">Prepare DESEQ2 Results<a class="anchor" aria-label="anchor" href="#prepare-deseq2-results"></a>
</h3>
<p>The test data that is provided with <code>SeedMatchR</code> was derived from the 2022 publication by Schlegel et al. The data set represents a DESeq2 analysis performed on rat liver that had been treated with Ttr targeting siRNA. We will use this example to explore seed mediated activity.</p>
<p>Notes: &gt;The <code>SeedMatchR</code> function will look for specific column in the input if using the <code>res</code> argument to map seed matches to differential expression data. The input must contain the columns <code>gene_id</code>, <code>log2FoldChange</code>, and <code>padj</code>.</p>
<div class="section level4">
<h4 id="download-data-only-need-to-perform-once-can-skip-to-loading-if-done">Download data (only need to perform once, can skip to loading if done)<a class="anchor" aria-label="anchor" href="#download-data-only-need-to-perform-once-can-skip-to-loading-if-done"></a>
</h4>
<p>We start by downloading the example data set. This function will download three files from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929" class="external-link">GSE184929</a>. These files represent three samples with different siRNA treatments at two dosages.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h4>
<p>We can load the example data into the environment.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sirna.data</span> <span class="op">=</span> <span class="fu"><a href="reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
<p>The DESeq2 results are available through the names <code>Schlegel_2022_Ttr_D1_30mkg</code>, <code>Schlegel_2022_Ttr_D4_30mkg</code> and <code>Schlegel_2022_Ttr_D1_10mkg</code>. The data set name is long, so it will be renamed to <code>res</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">sirna.data</span><span class="op">$</span><span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filter-example-results">Filter example results<a class="anchor" aria-label="anchor" href="#filter-example-results"></a>
</h4>
<p>The DESeq2 results file is then filtered. The function <code><a href="reference/filter_res.html">filter_res()</a></code> can be used to filter a results file by log2FoldChange, padj, baseMean, and remove NA entries.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dimensions before filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 32883    8</span></span>
<span><span class="co">#&gt; [1] 32883     8</span></span>
<span></span>
<span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="reference/filter_res.html">filter_res</a></span><span class="op">(</span><span class="va">res</span>, fdr_cutoff<span class="op">=</span><span class="fl">1</span>, fc_cutoff<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Dimensions after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 13582     8</span></span>
<span><span class="co">#&gt; [1] 8124    8</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="counting-seed-matches-in-transcripts">Counting seed matches in transcripts<a class="anchor" aria-label="anchor" href="#counting-seed-matches-in-transcripts"></a>
</h3>
<p>You can perform a seed match for a single seed using the <code><a href="reference/SeedMatchR.html">SeedMatchR()</a></code> function.</p>
<p>Notes:</p>
<blockquote>
<p>The names of the sequences in <code>seqs</code> will determine if you need to use the <code>tx.id.col</code> argument. If you sequence names are gene IDs, then no additional flags need to be set. If they sequence names are transcripts, then the argument <code>tx.id.col</code> should be set to <code>TRUE</code>. This will summarize the transcript matches to the gene level using information in the gtf file.</p>
</blockquote>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, </span>
<span>                 seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                 sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                 seed.name <span class="op">=</span> <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="comparing-the-expression-profiles-of-seed-targets-to-background">Comparing the expression profiles of seed targets to background<a class="anchor" aria-label="anchor" href="#comparing-the-expression-profiles-of-seed-targets-to-background"></a>
</h3>
<p>Many factors that perturb gene expression, like miRNA, show cumulative changes in their targets gene expression. Cumulative changes in the profile of genes expression can be visualized and tested with the emperical distribution function (ecdf) coupled with a statistical test such as the Kolmogorov-Smirnov test.</p>
<p><code>SeedMatchR</code> provides functions for comparing the log2(Fold Change) of two gene sets. The function <code>deseq_fc_ecdf</code> is designed to work directly with a DESeq2 results data frame.</p>
<p>Required Inputs:</p>
<ul>
<li>
<code>res</code>: DESeq2 results data frame</li>
<li>
<code>gene.lists</code>: A list of lists containing gene names</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene set 1 </span></span>
<span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Gene set 2</span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="reference/deseq_fc_ecdf.html">deseq_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, </span>
<span>                                  <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-13-1.png" width="50%"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=SeedMatchR" class="external-link">View on CRAN</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SeedMatchR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Tareian Cazares <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-4624-6156" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Gulcin Ozer <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8315-8051" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Pradeep Varathan Pugalenthi <br><small class="roles"> Author </small>   </li>
<li>Jibo Wang <br><small class="roles"> Author </small>   </li>
<li>Rick Higgs <br><small class="roles"> Author </small>   </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
