---
title: "Find Matches to siRNA in Genomic Features with SeedMatchR"
author: "Tareian Cazares, Pradeep Pugalenthi, Richard Higgs, Jibo Wang, and Hatice Gulcin Ozer"
vignette: >
  %\VignetteIndexEntry{SeedMatchR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
abstract: >
   On-target siRNA silencing of mRNA results from binding to
    fully complementary regions in transcripts to induce cleavage. 
    Off-target siRNA gene knockdown can occur through several modes, one being 
    a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated 
    off-target effects occur when the ~8 nucleotides at the 5’ end of the guide 
    strand, called a seed region, bind the 3’ untranslated regions of mRNA, 
    causing reduced translation. Experiments using siRNA knockdown paired with 
    RNA-seq can be used to detect siRNA sequences with potential off-target 
    effects driven by the seed region. 'SeedMatchR' provides tools for exploring 
    and detecting potential seed-mediated off-target effects of siRNA in RNA-seq 
    experiments. 'SeedMatchR' is designed to extend current differential 
    expression analysis tools, such as 'DESeq2', by annotating results with 
    predicted seed matches. Using publicly available data, we demonstrate the 
    ability of 'SeedMatchR' to detect cumulative changes in differential gene 
    expression attributed to siRNA seed regions.  
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = TRUE
)
```

## Installation

You can install the development version of SeedMatchR from [GitHub](https://github.com/) or the stable build from CRAN.

``` {r eval = FALSE}
# Install from GitHub
install.packages("devtools")
devtools::install_github("tacazares/SeedMatchR")
```

## Quick start example with public siRNA data

```{r include = FALSE}
# Import library
library(SeedMatchR)
library(GenomicFeatures)
```

### Introduction to example dataset

This example uses the siRNA sequence, D1, targeting the Ttr gene in rat liver from the publication:

```
Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga-Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp-Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside: Improving the clinical safety of GalNAc-siRNA conjugates using seed-pairing destabilization. Nucleic Acids Res. 2022 Jul 8;50(12):6656-6670. doi: 10.1093/nar/gkac539. PMID: 35736224; PMCID: PMC9262600.
```

The guide sequence of interest is 23 bp long and oriented 5' -> 3'. 

```{r}
# siRNA sequence of interest targeting a 23 bp region of the Ttr gene
guide.seq = "UUAUAGAGCAAGAACACUGUUUU"
```

### Run modes and input data

`SeedMatchR` can be run in two different modes. The first mode is designed for `DESEQ2` differential expression results and the second mode is designed for finding exact match positions regardless of differential expression data. 

* `DESEQ2` mode will map a column of match counts to a `data.frame` of differential expression resutls. 
* `data.frame` model will return a `data.frame` of matches that correspond to the interval ranges for matches in the input `DNAStringSet` library being searched. 

The required inputs to `SeedMatchR` are list below:

*  A character string representing the siRNA RNA sequence. This must be greater than 8 bp.
* `GTF`: gene transfer file containing species specific genomic information for gene bodies. This is used to derive the list of 3' UTRs and other features used in the analysis. This is also used to map transcript IDs to gene IDs.
* Feature-specific `DNAStringSet`: A `DNAStringSet` object of sequences for each of the features of interest. The features must be named according to the transcript they were derived from. Examples include those generated by `GenomicFeatures::extractTranscriptSeqs()` paired with functions like `GenomicFeatures::threeUTRsByTranscript()`.
* Seed name of the canonical seed you want to search or a custom definition with associated start and stop positions on the siRNA. 

Optional input: 

* `res`: A data frame of DESEQ2 results. If a DESEQ2 results are provided as a `data.frame`, then `SeedMatchR` will map that total match counts as a new column in the `data.frame`.

`SeedMatchR` provides functions for helping to wrangle the many data types and annotations. `SeedMatchR` makes extensive use of `AnnotationHub` to help access genomic annotations in a reproducible manner. 

#### Prepare species-specific annotation data

The function `load_annotations()` has built in annotation data for human, rat, and mouse annotations. More information can be found in [the vignette on annotations](articles/SeedMatchR_Annotation_Databases.Rmd). We can load the species specific annotations using the approach described below. Running this function will take several minutes. Therefore it might be helpful to save the objects and reload them later if you plan to use this code in a repeated workflow. 

```{r echo = T, results = 'hide', message=FALSE, warning=FALSE, error=FALSE}
# Load the species specific annotation database object
annodb = load_annotations(reference.name = "rnor6", canonical = FALSE, min.feature.width = 8, longest.utr = T)
```

#### Prepare DESEQ2 Results

The `DESEQ2` mode of `SeedMatchR` assumes that you will be performing your analysis on `DESEQ2` results outputs. The first step is to load your `DESEQ2` results file as a `data.frame`. 

The test data that is provided with `SeedMatchR` was derived from the 2022 publication by Schlegel et al. The data set represents a DESeq2 analysis performed on rat liver that had been treated with Ttr targeting siRNA. We will use this example to explore seed mediated activity. The data set name is long, so it will be renamed to `res`.

#### Download data (only need to perform once)

We start by downloading the example data set. This function will download three files from the GEO accession [GSE184929](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929). These files represent three samples with different siRNA treatments at two dosages. 

```{r}
get_example_data("sirna")
```

#### Load example data

We can load the example data into the environment.

```{r}
sirna.data = load_example_data("sirna")
```

The DESeq2 results are available through the names `Schlegel_2022_Ttr_D1_30mkg`, `Schlegel_2022_Ttr_D4_30mkg` and `Schlegel_2022_Ttr_D1_10mkg`. The data set name is long, so it will be renamed to `res`.

```{r}
res <- sirna.data$Schlegel_2022_Ttr_D1_30mkg
```

The `DESEQ2` results file is then filtered. The function `filter_res()` can be used to filter a results file by log2FoldChange, padj, baseMean, and remove NA entries. 

```{r}
# Dimensions before filtering

dim(res) # [1] 32883    6

# Filter DESeq2 results for SeedMatchR
res = filter_res(res)

# Dimensions after filtering
dim(res) # [1] 13582     8
```

### Get the seed sequence of interest

You can extract the seed sequence information from the siRNA input sequence using the `get_seed()` function. The inputs to the `get_seed()` function are the siRNA sequence of interest and the name of the seed.

```{r}
# Get the seed sequence information for the seed of interest
seed = get_seed(guide.seq = guide.seq, seed.name = "mer7m8")

seed
```

### Counting seed matches in transcripts

You can perform a seed match for a single seed using the `SeedMatchR()` function. 

```{r}
res = SeedMatchR(res = res, 
                 seqs = annodb$seqs, 
                 sequence = guide.seq, 
                 seed.name = "mer7m8")

head(res)
```

#### Match multiple seeds

You can perform seed matching for all available seeds using a for loop. The results will be appended as a new column to the results data frame. 

```{r}
for (seed in c("mer8", "mer6", "mer7A1")){
res = SeedMatchR(res = res, 
                 seqs = annodb$seqs, 
                 sequence = guide.seq, 
                 seed.name = seed)
}

head(res)
```

##### Match seeds with mismatches and indels allowed

You can also allow for inexact seed matches in your analysis with the `mismatches` and `indels` arguments. The names can be adjusted to reflect the arguments using the `col.name` argument. 

```{r}
  for (indel.bool in c(TRUE, FALSE)){
    for (mm in c(0,1)){
      for (seed in c("mer7m8", "mer8", "mer6", "mer7A1")){
        res <- SeedMatchR(res = res, 
                        seqs = annodb$seqs, 
                        sequence = guide.seq, 
                        seed.name = seed,
                        col.name = paste0(seed, 
                                          ".", 
                                          "mm", mm, 
                                          "_indel", indel.bool), 
                        max.mismatch = mm, 
                        with.indels = indel.bool)
      }
    }
  }

head(res)
```


### Returning match positions

#### Return as data frame

```{r}
match.df = SeedMatchR(seqs = annodb$seqs, 
                      sequence = guide.seq, 
                      seed.name = "mer7m8", 
                      res.format = "data.frame")

head(match.df)
```

#### Return as GRanges

```{r}
match.df = SeedMatchR(seqs = annodb$seqs, 
                      sequence = guide.seq, 
                      seed.name = "mer7m8", 
                      res.format = "granges")

head(match.df)
```

### Scan all sequences for all match types

```{r}
match.df = full_search(guide.seq = guide.seq, 
                       seqs = annodb$seqs, 
                       "Ttr")

head(match.df)
```

### Create a SeedMatchReport

```{r}
report = SeedMatchReport(res = res, 
                         seqs = annodb$seqs, 
                         group.name = "Ttr", 
                         guide.seq = guide.seq)

report$table
```

### Comparing the expression profiles of seed targets to background

Many factors that perturb gene expression, like miRNA, show cumulative changes in their targets gene expression. Cumulative changes in the profile of genes expression can be visualized and tested with the emperical distribution function (ecdf) coupled with a statistical test such as the Kolmogorov-Smirnov test. 

`SeedMatchR` provides functions for comparing the log2(Fold Change) of two gene sets. The function `deseq_fc_ecdf` is designed to work directly with a DESeq2 results data frame.

Required Inputs:

- `res`: `DESEQ2` results `data.frame`
- `gene.lists`: A list of lists containing gene names

```{r fig.height=5, fig.width=5, out.retina=1}
# Gene set 1 
mer7m8.list = res$gene_id[res$mer7m8 >= 1 & res$mer8 ==0]

# Gene set 2 
mer8.list = res$gene_id[res$mer8 >= 1]

background.list = res$gene_id[res$mer7m8 == 0 & res$mer8 == 0]

ecdf.results = deseq_fc_ecdf(res, 
                             title = "ECDF of log2(FC)",
                             list("Background" = background.list, 
                                  "mer8" = mer8.list, "mer7m8" = mer7m8.list),
                             stats_test = "KS", 
                             factor_order = c("Background", "mer8", "mer7m8"), 
                             null_name = "Background",
                             target_name = "mer8", 
                             alternative = "greater",
                             height = 5,
                             width = 5)

ecdf.results$plot
```

```{r}
sessionInfo() 
```
