<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot the ECDF of log2(fold change) between groups — deseq_fc_ecdf • SeedMatchR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the ECDF of log2(fold change) between groups — deseq_fc_ecdf"><meta name="description" content='This functions will take differential expression results as a
data.frame and plot the ECDF for the input gene_lists.
The gene sets to plot should be provided as a named list of vectors.
Example:
gene_lists = list("Background" = c("gene1", "gene2"), "Target" = c("gene2", "gene3"))
The output will be saved to a PDF if an output_filename is provided.
Users can define the groups that are to be compared in the statistical test
using the null_name and target_name arguments. The names must be found
in gene_lists. The factor_order is used to order the groups in the
analysis.
This functions returns:
$plot: The ECDF plot
$stats: The results object

'><meta property="og:description" content='This functions will take differential expression results as a
data.frame and plot the ECDF for the input gene_lists.
The gene sets to plot should be provided as a named list of vectors.
Example:
gene_lists = list("Background" = c("gene1", "gene2"), "Target" = c("gene2", "gene3"))
The output will be saved to a PDF if an output_filename is provided.
Users can define the groups that are to be compared in the statistical test
using the null_name and target_name arguments. The names must be found
in gene_lists. The factor_order is used to order the groups in the
analysis.
This functions returns:
$plot: The ECDF plot
$stats: The results object

'></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plot the ECDF of log2(fold change) between groups</h1>

      <div class="d-none name"><code>deseq_fc_ecdf.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This functions will take differential expression results as a
<code>data.frame</code> and plot the ECDF for the input <code>gene_lists</code>.</p>
<p>The gene sets to plot should be provided as a named list of vectors.</p>
<p>Example:</p>
<p><code>gene_lists = list("Background" = c("gene1", "gene2"), "Target" = c("gene2", "gene3"))</code></p>
<p>The output will be saved to a PDF if an <code>output_filename</code> is provided.</p>
<p>Users can define the groups that are to be compared in the statistical test
using the <code>null_name</code> and <code>target_name</code> arguments. The names must be found
in <code>gene_lists</code>. The <code>factor_order</code> is used to order the groups in the
analysis.</p>
<p>This functions returns:</p><ul><li><p><code>$plot</code>: The ECDF plot</p></li>
<li><p><code>$stats</code>: The results object</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">deseq_fc_ecdf</span><span class="op">(</span></span>
<span>  <span class="va">res</span>,</span>
<span>  <span class="va">gene_lists</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"ECDF"</span>,</span>
<span>  output_filename <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  palette <span class="op">=</span> <span class="va">SeedMatchR.palette</span>,</span>
<span>  factor_order <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  x_lims <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>  stats_test <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KS"</span>, <span class="st">"Wilcoxen"</span>, <span class="st">"Wass"</span>, <span class="st">"DTS"</span><span class="op">)</span>,</span>
<span>  alternative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"greater"</span>, <span class="st">"less"</span>, <span class="st">"two.sided"</span><span class="op">)</span>,</span>
<span>  null_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  target_name <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">320</span>,</span>
<span>  theme <span class="op">=</span> <span class="va">SeedMatchR.theme</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-res">res<a class="anchor" aria-label="anchor" href="#arg-res"></a></dt>
<dd><p>The differential expression results data frame.</p></dd>


<dt id="arg-gene-lists">gene_lists<a class="anchor" aria-label="anchor" href="#arg-gene-lists"></a></dt>
<dd><p>A nested list of gene names. Example:
gene_lists = list("Background" = background_list, "Target" = target_list)</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>The tile of the plot.</p></dd>


<dt id="arg-output-filename">output_filename<a class="anchor" aria-label="anchor" href="#arg-output-filename"></a></dt>
<dd><p>If the output filename is provided, then the plot is
saved.</p></dd>


<dt id="arg-palette">palette<a class="anchor" aria-label="anchor" href="#arg-palette"></a></dt>
<dd><p>Vector of colors to use for your curves.</p></dd>


<dt id="arg-factor-order">factor_order<a class="anchor" aria-label="anchor" href="#arg-factor-order"></a></dt>
<dd><p>Vector defining the order to use for the legends.</p></dd>


<dt id="arg-x-lims">x_lims<a class="anchor" aria-label="anchor" href="#arg-x-lims"></a></dt>
<dd><p>The x-axis range limits.</p></dd>


<dt id="arg-stats-test">stats_test<a class="anchor" aria-label="anchor" href="#arg-stats-test"></a></dt>
<dd><p>The statistical test to use. Options: KS, Wass, Wilcoxen, or
DTS.</p></dd>


<dt id="arg-alternative">alternative<a class="anchor" aria-label="anchor" href="#arg-alternative"></a></dt>
<dd><p>The alternative hypothesis to test. Options: greater,
less, two.sided.</p></dd>


<dt id="arg-null-name">null_name<a class="anchor" aria-label="anchor" href="#arg-null-name"></a></dt>
<dd><p>The name in the gene.list to use as the null for ECDF plots.</p></dd>


<dt id="arg-target-name">target_name<a class="anchor" aria-label="anchor" href="#arg-target-name"></a></dt>
<dd><p>The name in the gene.list to use as the target for ECDF.</p></dd>


<dt id="arg-height">height<a class="anchor" aria-label="anchor" href="#arg-height"></a></dt>
<dd><p>Plot height in inches.</p></dd>


<dt id="arg-width">width<a class="anchor" aria-label="anchor" href="#arg-width"></a></dt>
<dd><p>Plot width in inches.</p></dd>


<dt id="arg-dpi">dpi<a class="anchor" aria-label="anchor" href="#arg-dpi"></a></dt>
<dd><p>The DPI resolution for the figure.</p></dd>


<dt id="arg-theme">theme<a class="anchor" aria-label="anchor" href="#arg-theme"></a></dt>
<dd><p>A ggplot2 theme defining plot parameters.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot object and ECDF statistics.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># interactive()</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">anno.db</span> <span class="op">=</span> <span class="fu"><a href="load_annotations.html">load_annotations</a></span><span class="op">(</span><span class="st">"rnor7"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load test data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sirna.data</span> <span class="op">=</span> <span class="fu"><a href="load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">sirna.data</span><span class="op">$</span><span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="filter_res.html">filter_res</a></span><span class="op">(</span><span class="va">res</span>, fdr.cutoff<span class="op">=</span><span class="fl">1</span>, fc.cutoff<span class="op">=</span><span class="fl">0</span>, rm.na.log2fc <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, gtf <span class="op">=</span> <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, seqs <span class="op">=</span> <span class="va">anno.db</span><span class="op">$</span><span class="va">seqs</span>,</span></span>
<span class="r-in"><span>sequence <span class="op">=</span> <span class="va">guide.seq</span>, seed.name <span class="op">=</span> <span class="st">"mer7m8"</span>, tx.id.col<span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Gene set 1</span></span></span>
<span class="r-in"><span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Gene set 2</span></span></span>
<span class="r-in"><span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu">deseq_fc_ecdf</span><span class="op">(</span><span class="va">res</span>,</span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

