<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ SeedMatchR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="SeedMatchR.html">SeedMatchR()</a></code>

        </dt>
        <dd>Find seed matches in genomic features</dd>
      </dl><dl><dt>

          <code><a href="SeedMatchReport.html">SeedMatchReport()</a></code>

        </dt>
        <dd>Generate statistics based on sequence matches and DE data</dd>
      </dl><dl><dt>

          <code><a href="as_dataframe_mp_matches.html">as_dataframe_mp_matches()</a></code>

        </dt>
        <dd>as_dataframe_mp_matches</dd>
      </dl><dl><dt>

          <code><a href="benchmark_preds.html">benchmark_preds()</a></code>

        </dt>
        <dd>Benchmark seed match predictions</dd>
      </dl><dl><dt>

          <code><a href="build_annotation_filter.html">build_annotation_filter()</a></code>

        </dt>
        <dd>Build an <a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilter.html" class="external-link">AnnotationFilter::AnnotationFilter</a> for selecting transcripts</dd>
      </dl><dl><dt>

          <code><a href="check_bulges_nt.html">check_bulges_nt()</a></code>

        </dt>
        <dd>check_bulges_nt</dd>
      </dl><dl><dt>

          <code><a href="create_graph.html">create_graph()</a></code>

        </dt>
        <dd>Create a graph of protein-protein interactions from stringDB</dd>
      </dl><dl><dt>

          <code><a href="deseq_fc_ecdf.html">deseq_fc_ecdf()</a></code>

        </dt>
        <dd>Plot the ECDF of log2(fold change) between groups</dd>
      </dl><dl><dt>

          <code><a href="dot-check_gene_list_overlap.html">.check_gene_list_overlap()</a></code>

        </dt>
        <dd>Check if input gene lists overlap</dd>
      </dl><dl><dt>

          <code><a href="dot-create_alignment_df.html">.create_alignment_df()</a></code>

        </dt>
        <dd>Create a dataframe of alignment information</dd>
      </dl><dl><dt>

          <code><a href="dot-find_factorial_internal.html">.find_factorial_internal()</a></code>

        </dt>
        <dd>find_factorial_internal</dd>
      </dl><dl><dt>

          <code><a href="dot-gc_content.html">.gc_content()</a></code>

        </dt>
        <dd>.gc_content</dd>
      </dl><dl><dt>

          <code><a href="dot-get_all_combination_w_b.html">.get_all_combination_w_b()</a></code>

        </dt>
        <dd>get_all_combination_w_b</dd>
      </dl><dl><dt>

          <code><a href="dot-get_canonical_seed.html">.get_canonical_seed()</a></code>

        </dt>
        <dd>Get the target sequence for canonical seed definitions</dd>
      </dl><dl><dt>

          <code><a href="dot-get_counts_column.html">.get_counts_column()</a></code>

        </dt>
        <dd>Get match results as a column of counts appended to a differential expression results <code>data.frame</code></dd>
      </dl><dl><dt>

          <code><a href="dot-get_custom_sequence.html">.get_custom_sequence()</a></code>

        </dt>
        <dd>Get a custom target sequence based on start and stop position</dd>
      </dl><dl><dt>

          <code><a href="dot-get_longest_utr.html">.get_longest_utr()</a></code>

        </dt>
        <dd>Get the longest transcript per gene</dd>
      </dl><dl><dt>

          <code><a href="dot-get_match_df.html">.get_match_df()</a></code>

        </dt>
        <dd>Get sequence matches as a <code>data.frame</code></dd>
      </dl><dl><dt>

          <code><a href="dot-get_target_bulged_guides.html">.get_target_bulged_guides()</a></code>

        </dt>
        <dd>get_target_bulged_guides</dd>
      </dl><dl><dt>

          <code><a href="dot-make_space.html">.make_space()</a></code>

        </dt>
        <dd>Create a vector of spaces for padding sequences for alignment</dd>
      </dl><dl><dt>

          <code><a href="dot-matchStrings.html">.matchStrings()</a></code>

        </dt>
        <dd>Match two strings of the same size and return a character vector to represent positions that match</dd>
      </dl><dl><dt>

          <code><a href="dot-plot_ecdf.html">.plot_ecdf()</a></code>

        </dt>
        <dd>Use ggplot2 to plot the ecdf for log2 fold changes</dd>
      </dl><dl><dt>

          <code><a href="dot-reduce_features.html">.reduce_features()</a></code>

        </dt>
        <dd>Reduce 3' UTRs to the gene level across all transcripts</dd>
      </dl><dl><dt>

          <code><a href="dot-sequence_with_tb.html">.sequence_with_tb()</a></code>

        </dt>
        <dd>.sequence_with_tb</dd>
      </dl><dl><dt>

          <code><a href="dot-update_sequence_at.html">.update_sequence_at()</a></code>

        </dt>
        <dd>.update_sequence_at</dd>
      </dl><dl><dt>

          <code><a href="download_parse_file.html">download_parse_file()</a></code>

        </dt>
        <dd>Download and parse DESeq2 output from GSE184929</dd>
      </dl><dl><dt>

          <code><a href="ecdf_stat_test.html">ecdf_stat_test()</a></code>

        </dt>
        <dd>Test for differences in log2(Fold Change) ECDFs between two gene lists</dd>
      </dl><dl><dt>

          <code><a href="filter_res.html">filter_res()</a></code>

        </dt>
        <dd>Filter differential expression results</dd>
      </dl><dl><dt>

          <code><a href="full_search.html">full_search()</a></code>

        </dt>
        <dd>Search input <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a></code> for a set of defined sequences</dd>
      </dl><dl><dt>

          <code><a href="full_search_ranges.html">full_search_ranges()</a></code>

        </dt>
        <dd>Search input <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a></code> for a set of defined sequences</dd>
      </dl><dl><dt>

          <code><a href="generate_e_value.html">generate_e_value()</a></code>

        </dt>
        <dd>generate_e_value</dd>
      </dl><dl><dt>

          <code><a href="get_3UTR_granges.html">get_3UTR_granges()</a></code>

        </dt>
        <dd>Extract <a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GenomicRanges::GRangesList</a> of 3' UTRs</dd>
      </dl><dl><dt>

          <code><a href="get_compensatory_sequences.html">get_compensatory_sequences()</a></code>

        </dt>
        <dd>get_compensatory_sequences</dd>
      </dl><dl><dt>

          <code><a href="get_example_data.html">get_example_data()</a></code>

        </dt>
        <dd>Download example DESeq2 data from GEO</dd>
      </dl><dl><dt>

          <code><a href="get_network_matches.html">get_network_matches()</a></code>

        </dt>
        <dd>get_network_matches</dd>
      </dl><dl><dt>

          <code><a href="get_seed.html">get_seed()</a></code>

        </dt>
        <dd>Get the target sequence for a given siRNA input sequence</dd>
      </dl><dl><dt>

          <code><a href="get_stringPPI_to_stringGGI.html">get_stringPPI_to_stringGGI()</a></code>

        </dt>
        <dd>get_stringPPI_to_stringGGI</dd>
      </dl><dl><dt>

          <code><a href="kmer_hist.html">kmer_hist()</a></code>

        </dt>
        <dd>Plot a histogram of kmer frequency across genes</dd>
      </dl><dl><dt>

          <code><a href="load_annotations.html">load_annotations()</a></code>

        </dt>
        <dd>Load species-specific annotations needed to run <code><a href="../reference/SeedMatchR.html">SeedMatchR()</a></code></dd>
      </dl><dl><dt>

          <code><a href="load_example_data.html">load_example_data()</a></code>

        </dt>
        <dd>Load example DESeq2 data into the environment</dd>
      </dl><dl><dt>

          <code><a href="load_species_anno_db.html">load_species_anno_db()</a></code>

        </dt>
        <dd>Load species-specific annotations from <code>AnnotationHub</code></dd>
      </dl><dl><dt>

          <code><a href="match_distances.html">match_distances()</a></code>

        </dt>
        <dd>Find inter match distances for each gene</dd>
      </dl><dl><dt>

          <code><a href="message_passing.html">message_passing()</a></code>

        </dt>
        <dd>message_passing</dd>
      </dl><dl><dt>

          <code><a href="plotAlignment.html">plotAlignment()</a></code>

        </dt>
        <dd>Plot matches as alignments against target sequence feature</dd>
      </dl><dl><dt>

          <code><a href="run_wobble_bulges_pipeline.html">run_wobble_bulges_pipeline()</a></code>

        </dt>
        <dd>Running wobble and bulge inclusive seedmatchR pipeline</dd>
      </dl><dl><dt>

          <code><a href="sequence_kmer_counts.html">sequence_kmer_counts()</a></code>

        </dt>
        <dd>Count the k-mer frequency in a set of sequences</dd>
      </dl><dl><dt>

          <code><a href="vmatchPattern2.html">vmatchPattern2()</a></code>

        </dt>
        <dd>A modified vmatchPattern function that supports indels and mismatches</dd>
      </dl><dl><dt>

          <code><a href="wass_dist.html">wass_dist()</a></code>

        </dt>
        <dd>Function to compute overall, positive, and negative Wasserstein differences between ECDFs</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

