<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Load species-specific annotations needed to run SeedMatchR() — load_annotations • SeedMatchR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Load species-specific annotations needed to run SeedMatchR() — load_annotations"><meta name="description" content="load_annotations uses AnnotationHub::AnnotationHub() to load
species-specific annotations. This function currently works for human, rat,
and mouse."><meta property="og:description" content="load_annotations uses AnnotationHub::AnnotationHub() to load
species-specific annotations. This function currently works for human, rat,
and mouse."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Load species-specific annotations needed to run <code><a href="SeedMatchR.html">SeedMatchR()</a></code></h1>

      <div class="d-none name"><code>load_annotations.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>load_annotations</code> uses <code><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub::AnnotationHub()</a></code> to load
species-specific annotations. This function currently works for human, rat,
and mouse.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">load_annotations</span><span class="op">(</span></span>
<span>  reference.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg38"</span>, <span class="st">"hg38-old"</span>, <span class="st">"mm39"</span>, <span class="st">"mm10"</span>, <span class="st">"rnor6"</span>, <span class="st">"rnor7"</span><span class="op">)</span>,</span>
<span>  feature.type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3UTR"</span>, <span class="st">"5UTR"</span>, <span class="st">"exons"</span>, <span class="st">"cds"</span><span class="op">)</span>,</span>
<span>  standard.chroms <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove.na.rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  protein.coding <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  transcript.support <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  canonical <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tx_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  symbol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  entrez_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  longest.utr <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  reduce.features <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  add.filter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.feature.width <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  return_gene_name <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-reference-name">reference.name<a class="anchor" aria-label="anchor" href="#arg-reference-name"></a></dt>
<dd><p>Reference build name. Options: hg38, rnor6, rnor7,
mm39, mm10</p></dd>


<dt id="arg-feature-type">feature.type<a class="anchor" aria-label="anchor" href="#arg-feature-type"></a></dt>
<dd><p>The transcript feature type to extract. Options: "3UTR", "5UTR", "exons", "cds"</p></dd>


<dt id="arg-standard-chroms">standard.chroms<a class="anchor" aria-label="anchor" href="#arg-standard-chroms"></a></dt>
<dd><p>Keep only standard chromosomes. Boolean.</p></dd>


<dt id="arg-remove-na-rows">remove.na.rows<a class="anchor" aria-label="anchor" href="#arg-remove-na-rows"></a></dt>
<dd><p>Remove transcripts with NA in the ID column. Boolean.</p></dd>


<dt id="arg-protein-coding">protein.coding<a class="anchor" aria-label="anchor" href="#arg-protein-coding"></a></dt>
<dd><p>Keep only protein coding genes. Boolean.</p></dd>


<dt id="arg-transcript-support">transcript.support<a class="anchor" aria-label="anchor" href="#arg-transcript-support"></a></dt>
<dd><p>Filter by transcript support level. Integer 1-4</p></dd>


<dt id="arg-canonical">canonical<a class="anchor" aria-label="anchor" href="#arg-canonical"></a></dt>
<dd><p>Keep only the ENSEMBL canonical transcript. Boolean.</p></dd>


<dt id="arg-gene-id">gene_id<a class="anchor" aria-label="anchor" href="#arg-gene-id"></a></dt>
<dd><p>Keep specific ENSEMBL ID(s). Input is a vector</p></dd>


<dt id="arg-tx-id">tx_id<a class="anchor" aria-label="anchor" href="#arg-tx-id"></a></dt>
<dd><p>Keep specific tx_id(s). Input is a vector</p></dd>


<dt id="arg-symbol">symbol<a class="anchor" aria-label="anchor" href="#arg-symbol"></a></dt>
<dd><p>Keep specific gene symbol(s). Input is a vector</p></dd>


<dt id="arg-entrez-id">entrez_id<a class="anchor" aria-label="anchor" href="#arg-entrez-id"></a></dt>
<dd><p>Keep specific ENTREZ ID(s). Input is a vector</p></dd>


<dt id="arg-longest-utr">longest.utr<a class="anchor" aria-label="anchor" href="#arg-longest-utr"></a></dt>
<dd><p>Reduce annotations to the longest UTR per gene</p></dd>


<dt id="arg-reduce-features">reduce.features<a class="anchor" aria-label="anchor" href="#arg-reduce-features"></a></dt>
<dd><p>Reduce 3' UTRs to non-overlapping intervals across all
transcripts per gene.</p></dd>


<dt id="arg-add-filter">add.filter<a class="anchor" aria-label="anchor" href="#arg-add-filter"></a></dt>
<dd><p>Optional <a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilterList.html" class="external-link">AnnotationFilter::AnnotationFilterList</a> to include</p></dd>


<dt id="arg-min-feature-width">min.feature.width<a class="anchor" aria-label="anchor" href="#arg-min-feature-width"></a></dt>
<dd><p>The minimum length for features in nucleotides.</p></dd>


<dt id="arg-return-gene-name">return_gene_name<a class="anchor" aria-label="anchor" href="#arg-return-gene-name"></a></dt>
<dd><p>If true, the gene_id will be used to name the sequences.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The following are returned in the object:</p><ul><li><p><code>$gtf</code>: A <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a> object containing the transcript information</p></li>
<li><p><code>$txdb</code>: A <a href="https://rdrr.io/pkg/ensembldb/man/EnsDb.html" class="external-link">ensembldb::EnsDb</a> object of transcript information</p></li>
<li><p><code>$dna</code>: The <a href="https://rdrr.io/pkg/rtracklayer/man/TwoBitFile-class.html" class="external-link">rtracklayer::TwoBitFile</a> DNA sequence</p></li>
<li><p><code>$features</code>: A <a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GenomicRanges::GRangesList</a> object representing the features of interest</p></li>
<li><p><code>$seqs</code>: A <a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a> for the features in <code>$features</code></p></li>
<li><p><code>$filter</code>: A <a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilterList.html" class="external-link">AnnotationFilter::AnnotationFilterList</a> used for filtering the <a href="https://rdrr.io/pkg/ensembldb/man/EnsDb.html" class="external-link">ensembldb::EnsDb</a> object</p></li>
<li><p><code>$tx.names</code>: A list of transcript names in <code>$seqs</code></p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function is designed to perform all annotation loading, parsing, and
sequence extractions for the reference of interest.</p>
<p><code>load_annotations</code> uses the <code>ensembldb</code> package for handling annotations.
This allows for the use of <a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilter.html" class="external-link">AnnotationFilter::AnnotationFilter</a>s for
easily selecting transcripts of interest.</p>
<p>Most of the arguments to <code>load_annotations</code> are geared towards customizing
the output features and sequence extracted from the <a href="https://rdrr.io/pkg/ensembldb/man/EnsDb.html" class="external-link">ensembldb::EnsDb</a>
object. Filtering parameters are inherited from
<code><a href="build_annotation_filter.html">build_annotation_filter()</a></code>`.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># interactive()</span></span></span>
<span class="r-in"><span><span class="va">anno.db</span> <span class="op">=</span> <span class="fu">load_annotations</span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

