<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Find seed matches in genomic features — SeedMatchR • SeedMatchR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find seed matches in genomic features — SeedMatchR"><meta name="description" content="Find seed matches in a Biostrings::DNAStringSet object of
sequences.
This function will use get.seed to extract the seed sequence
from the guide sequence.
There are two modes for running SeedMatchR. The different modes ares set by
the res.format argument with the options DESEQ2 or data.frame.
The DESEQ2 mode will match the input seed across all rows of the
Biostrings::DNAStringSet object using Biostrings::vcountPattern(). These matches are then
aggregated and matched to genes in the DESEQ2 results data.frame. Counts
are reported as an additional column with the seed count. By default, this
mode will be run if a DESEQ2 results data.frame is provided to the res argument.
The data.frame mode will return a data.frame with the interval ranges
associated with matches from Biostrings::vmatchPattern(). This is the default mode run
when no DESEQ2 results are provided."><meta property="og:description" content="Find seed matches in a Biostrings::DNAStringSet object of
sequences.
This function will use get.seed to extract the seed sequence
from the guide sequence.
There are two modes for running SeedMatchR. The different modes ares set by
the res.format argument with the options DESEQ2 or data.frame.
The DESEQ2 mode will match the input seed across all rows of the
Biostrings::DNAStringSet object using Biostrings::vcountPattern(). These matches are then
aggregated and matched to genes in the DESEQ2 results data.frame. Counts
are reported as an additional column with the seed count. By default, this
mode will be run if a DESEQ2 results data.frame is provided to the res argument.
The data.frame mode will return a data.frame with the interval ranges
associated with matches from Biostrings::vmatchPattern(). This is the default mode run
when no DESEQ2 results are provided."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find seed matches in genomic features</h1>

      <div class="d-none name"><code>SeedMatchR.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Find seed matches in a <a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a> object of
sequences.</p>
<p>This function will use <code>get.seed</code> to extract the seed sequence
from the guide sequence.</p>
<p>There are two modes for running <code>SeedMatchR</code>. The different modes ares set by
the <code>res.format</code> argument with the options <code>DESEQ2</code> or <code>data.frame</code>.</p>
<p>The <code>DESEQ2</code> mode will match the input seed across all rows of the
<a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a> object using <code><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html" class="external-link">Biostrings::vcountPattern()</a></code>. These matches are then
aggregated and matched to genes in the <code>DESEQ2</code> results <code>data.frame</code>. Counts
are reported as an additional column with the seed count. By default, this
mode will be run if a <code>DESEQ2</code> results <code>data.frame</code> is provided to the <code>res</code> argument.</p>
<p>The <code>data.frame</code> mode will return a <code>data.frame</code> with the interval ranges
associated with matches from <code><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html" class="external-link">Biostrings::vmatchPattern()</a></code>. This is the default mode run
when no <code>DESEQ2</code> results are provided.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SeedMatchR</span><span class="op">(</span></span>
<span>  <span class="va">seqs</span>,</span>
<span>  <span class="va">sequence</span>,</span>
<span>  res <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  res.format <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DESEQ2"</span>, <span class="st">"data.frame"</span>, <span class="st">"granges"</span>, <span class="st">"iranges"</span><span class="op">)</span>,</span>
<span>  seed.name <span class="op">=</span> <span class="st">"mer7m8"</span>,</span>
<span>  col.name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  start.pos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stop.pos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  match.df <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sirna.name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  shared_genes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  allow_wobbles <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  get_seed <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  max.mismatch <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  with.indels <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fixed <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seqs">seqs<a class="anchor" aria-label="anchor" href="#arg-seqs"></a></dt>
<dd><p>The <a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a> object with sequence information
for features. The names of the sequences should be the transcript names.</p></dd>


<dt id="arg-sequence">sequence<a class="anchor" aria-label="anchor" href="#arg-sequence"></a></dt>
<dd><p>The <a href="https://rdrr.io/pkg/Biostrings/man/DNAString-class.html" class="external-link">Biostrings::DNAString</a> guide sequence oriented 5' &gt; 3'.</p></dd>


<dt id="arg-res">res<a class="anchor" aria-label="anchor" href="#arg-res"></a></dt>
<dd><p>An optional <code>DESEQ2</code> results <code>data.frame</code>. If provided, an
additional column with the seed match count will be added to the
<code>data.frame</code>. If not provided, <code>SeedMatchR</code> will return the interval ranges
for each match for the input <a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a>.</p></dd>


<dt id="arg-res-format">res.format<a class="anchor" aria-label="anchor" href="#arg-res-format"></a></dt>
<dd><p>Format for the returned results. Either 'data.frame', 'DESEQ2', 'iranges', or 'granges'</p></dd>


<dt id="arg-seed-name">seed.name<a class="anchor" aria-label="anchor" href="#arg-seed-name"></a></dt>
<dd><p>The seed name to be reported in the <code>data.frame</code> column
called <code>seed</code></p></dd>


<dt id="arg-col-name">col.name<a class="anchor" aria-label="anchor" href="#arg-col-name"></a></dt>
<dd><p>The optional name of the column of match counts. Will default
to <code>seed.name</code> if not set</p></dd>


<dt id="arg-start-pos">start.pos<a class="anchor" aria-label="anchor" href="#arg-start-pos"></a></dt>
<dd><p>The seed start position</p></dd>


<dt id="arg-stop-pos">stop.pos<a class="anchor" aria-label="anchor" href="#arg-stop-pos"></a></dt>
<dd><p>The seed stop position</p></dd>


<dt id="arg-match-df">match.df<a class="anchor" aria-label="anchor" href="#arg-match-df"></a></dt>
<dd><p>Optional: If a matches df is provided the results of the
current search will be added with rbind.</p></dd>


<dt id="arg-sirna-name">sirna.name<a class="anchor" aria-label="anchor" href="#arg-sirna-name"></a></dt>
<dd><p>Optional siRNA name. A new column called <code>sirna.name</code> will be added to <code>data.frame</code></p></dd>


<dt id="arg-shared-genes">shared_genes<a class="anchor" aria-label="anchor" href="#arg-shared-genes"></a></dt>
<dd><p>If true, the tx set is reduced to overlapping
features in both the sequences db and in the DE data.</p></dd>


<dt id="arg-allow-wobbles">allow_wobbles<a class="anchor" aria-label="anchor" href="#arg-allow-wobbles"></a></dt>
<dd><p>If True, allow G:U wobbles by replacing U with Y.</p></dd>


<dt id="arg-get-seed">get_seed<a class="anchor" aria-label="anchor" href="#arg-get-seed"></a></dt>
<dd><p>If True, parse the input character vector and return get_seed object.</p></dd>


<dt id="arg-max-mismatch">max.mismatch<a class="anchor" aria-label="anchor" href="#arg-max-mismatch"></a></dt>
<dd><p>Number of allowed mismatches or the total edit distance</p></dd>


<dt id="arg-with-indels">with.indels<a class="anchor" aria-label="anchor" href="#arg-with-indels"></a></dt>
<dd><p>If True, include indels</p></dd>


<dt id="arg-fixed">fixed<a class="anchor" aria-label="anchor" href="#arg-fixed"></a></dt>
<dd><p>Require that each sequence symbol matches when searching. Should be FALSE if using wobbles.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Either a 'data.frame', 'DESEQ2', 'iranges', or 'granges' object based on res.format input</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># interactive()</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">anno.db</span> <span class="op">=</span> <span class="fu"><a href="load_annotations.html">load_annotations</a></span><span class="op">(</span><span class="st">"rnor7"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sirna.data</span> <span class="op">=</span> <span class="fu"><a href="load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">sirna.data</span><span class="op">$</span><span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="filter_res.html">filter_res</a></span><span class="op">(</span><span class="va">res</span>, fdr.cutoff<span class="op">=</span><span class="fl">1</span>, fc.cutoff<span class="op">=</span><span class="fl">0</span>, rm.na.log2fc <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">=</span> <span class="fu">SeedMatchR</span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, seqs <span class="op">=</span> <span class="va">anno.db</span><span class="op">$</span><span class="va">seqs</span>,</span></span>
<span class="r-in"><span>sequence <span class="op">=</span> <span class="va">guide.seq</span>, seed.name <span class="op">=</span> <span class="st">"mer7m8"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

