<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SeedMatchR">
<title>Find Matches to Canonical siRNA Seeds in Genomic Features with SeedMatchR • SeedMatchR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Matches to Canonical siRNA Seeds in Genomic Features with SeedMatchR">
<meta property="og:description" content="SeedMatchR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/SeedMatchR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Find Matches to Canonical siRNA Seeds in Genomic Features with SeedMatchR</h1>
                        <h4 data-toc-skip class="author">Tareian Cazares
and Gulcin Ozer</h4>
            
      
      
      <div class="d-none name"><code>SeedMatchR.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>On-target gene knockdown using siRNA ideally results from
      binding fully complementary regions in mRNA transcripts to induce
      cleavage. Off-target siRNA gene knockdown can occur through
      several modes, one being a seed-mediated mechanism mimicking miRNA
      gene regulation. Seed-mediated off-target effects occur when the
      ~8 nucleotides at the 5’ end of the guide strand, called a seed
      region, bind the 3’ untranslated regions of mRNA, causing reduced
      translation. Experiments using siRNA knockdown paired with RNA-seq
      can be used to detect siRNA sequences with potential off-target
      effects driven by the seed region. SeedMatchR provides tools for
      exploring and detecting potential seed-mediated off-target effects
      of siRNA in RNA-seq experiments. SeedMatchR is designed to extend
      current differential expression analysis tools, such as DESeq2, by
      annotating results with predicted seed matches. Using publicly
      available data, we demonstrate the ability of SeedMatchR to detect
      cumulative changes in differential gene expression attributed to
      siRNA seed regions.</p>
    </div>
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of SeedMatchR from <a href="https://github.com/" class="external-link">GitHub</a> or the stable build from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tacazares/SeedMatchR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start-example-with-public-sirna-data">Quick start example with public siRNA data<a class="anchor" aria-label="anchor" href="#quick-start-example-with-public-sirna-data"></a>
</h2>
<div class="section level3">
<h3 id="introduction-to-example-dataset">Introduction to example dataset<a class="anchor" aria-label="anchor" href="#introduction-to-example-dataset"></a>
</h3>
<p>This example uses the siRNA sequence, D1, targeting the Ttr gene in
rat liver from the publication:</p>
<pre><code>Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga-Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp-Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside: Improving the clinical safety of GalNAc-siRNA conjugates using seed-pairing destabilization. Nucleic Acids Res. 2022 Jul 8;50(12):6656-6670. doi: 10.1093/nar/gkac539. PMID: 35736224; PMCID: PMC9262600.</code></pre>
<p>The guide sequence of interest is 23 bp long and oriented 5’ -&gt;
3’.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># siRNA sequence of interest targeting a 23 bp region of the Ttr gene</span></span>
<span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-input-data">Required Input Data<a class="anchor" aria-label="anchor" href="#required-input-data"></a>
</h3>
<p>The required inputs to <code>SeedMatchR</code> are a DESeq2 results
data frame in addition to species specific annoation data such as GTF,
2bit DNA, and 3’ UTR sequences.</p>
<p><code>SeedMatchR</code> makes extensive use of
<code>AnnotationDB</code> objects to help access genomic information in
a reproducible manner. The required annotations are:</p>
<ul>
<li>A character string representing the siRNA RNA sequence. This must be
greater than 8 bp.</li>
<li>
<code>res</code>: a data frame of DESEQ2 results.</li>
<li>
<code>GTF</code>: gene transfer file containing species specific
genomic information for gene bodies. This is used to derive the list of
3’ UTRs and other features used in the analysis. This is also used to
map transcript IDs to gene IDs.</li>
<li>Feature-specific <code>DNAStringSet</code>: A
<code>DNAStringSet</code> object of sequences for each of the features
of interest. The features must be named according to the transcript they
were derived from. Examples include those generated by
<code><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html" class="external-link">GenomicFeatures::extractTranscriptSeqs()</a></code> paired with
functions like
<code><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html" class="external-link">GenomicFeatures::threeUTRsByTranscript()</a></code>.</li>
</ul>
<div class="section level4">
<h4 id="prepare-species-specific-annotation-data">Prepare species-specific annotation data<a class="anchor" aria-label="anchor" href="#prepare-species-specific-annotation-data"></a>
</h4>
<p>The function <code><a href="../reference/load_species_anno_db.html">load_species_anno_db()</a></code> has built in
annotation data for human, rat, and mouse annotations. We can load the
species specific annotations using the following approach:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the species specific annotation database object</span></span>
<span><span class="va">anno.db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_species_anno_db.html">load_species_anno_db</a></span><span class="op">(</span><span class="st">"rat"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="extract-features-and-sequences-of-interest-from-annotations">Extract features and sequences of interest from annotations<a class="anchor" aria-label="anchor" href="#extract-features-and-sequences-of-interest-from-annotations"></a>
</h4>
<p>We will use the annotations to derive the features and feature
sequences that we want to scan for each gene.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">features</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_feature_seqs.html">get_feature_seqs</a></span><span class="op">(</span><span class="va">anno.db</span><span class="op">$</span><span class="va">tx.db</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">dna</span>, feature.type <span class="op">=</span> <span class="st">"3UTR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="prepare-deseq2-results">Prepare DESEQ2 Results<a class="anchor" aria-label="anchor" href="#prepare-deseq2-results"></a>
</h4>
<p><code>SeedMatchR</code> assumes that you will be performing your
analysis on DESEQ2 results outputs. The first step is to load your
DESEQ2 results file as a data frame.</p>
<p>The test data that is provided with <code>SeedMatchR</code> was
derived from the 2022 publication by Schlegel et al. The data set
represents a DESeq2 analysis performed on rat liver that had been
treated with Ttr targeting siRNA. We will use this example to explore
seed mediated activity. The data set name is long, so it will be renamed
to <code>res</code>.</p>
</div>
<div class="section level4">
<h4 id="download-data-only-need-to-perform-once">Download data (only need to perform once)<a class="anchor" aria-label="anchor" href="#download-data-only-need-to-perform-once"></a>
</h4>
<p>We start by downloading the example data set. This function will
download three files from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929" class="external-link">GSE184929</a>.
These files represent three samples with different siRNA treatments at
two dosages.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h4>
<p>We can load the example data into the environment.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
<p>The DESeq2 results are available through the names
<code>Schlegel_2022_Ttr_D1_30mkg</code>,
<code>Schlegel_2022_Ttr_D4_30mkg</code> and
<code>Schlegel_2022_Ttr_D1_10mkg</code>. The data set name is long, so
it will be renamed to <code>res</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></code></pre></div>
<p>The DESeq2 results file is then filtered. The function
<code><a href="../reference/filter_deseq.html">filter_deseq()</a></code> can be used to filter a results file by
log2FoldChange, padj, baseMean, and remove NA entries.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dimensions before filtering</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 32883    6</span></span>
<span><span class="co">#&gt; [1] 32883     8</span></span>
<span></span>
<span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_deseq.html">filter_deseq</a></span><span class="op">(</span><span class="va">res</span>, fdr.cutoff<span class="op">=</span><span class="fl">1</span>, fc.cutoff<span class="op">=</span><span class="fl">0</span>, rm.na.log2fc <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Dimensions after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 13582     8</span></span>
<span><span class="co">#&gt; [1] 13582     8</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="plot-possible-seeds">Plot possible seeds<a class="anchor" aria-label="anchor" href="#plot-possible-seeds"></a>
</h3>
<p>Use the <code><a href="../reference/plot_seeds.html">plot_seeds()</a></code> function to visualize the available
SeedMatchR options for your input sequence of interest. The only input
to <code><a href="../reference/plot_seeds.html">plot_seeds()</a></code> is your input siRNA sequence of interest.
This function assumes that you are using a RNA input.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the seed sequence options for the siRNA of interest</span></span>
<span><span class="va">avail.seed.plot</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_seeds.html">plot_seeds</a></span><span class="op">(</span><span class="va">guide.seq</span><span class="op">)</span></span>
<span><span class="co">#&gt; use default substitution matrix</span></span>
<span><span class="co">#&gt; Registered S3 methods overwritten by 'ggalt':</span></span>
<span><span class="co">#&gt;   method                  from   </span></span>
<span><span class="co">#&gt;   grid.draw.absoluteGrob  ggplot2</span></span>
<span><span class="co">#&gt;   grobHeight.absoluteGrob ggplot2</span></span>
<span><span class="co">#&gt;   grobWidth.absoluteGrob  ggplot2</span></span>
<span><span class="co">#&gt;   grobX.absoluteGrob      ggplot2</span></span>
<span><span class="co">#&gt;   grobY.absoluteGrob      ggplot2</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span></span>
<span><span class="va">avail.seed.plot</span></span></code></pre></div>
<p><img src="SeedMatchR_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="get-the-seed-sequence-of-interest">Get the seed sequence of interest<a class="anchor" aria-label="anchor" href="#get-the-seed-sequence-of-interest"></a>
</h3>
<p>You can extract the seed sequence information from the siRNA input
sequence using the <code><a href="../reference/get_seed.html">get_seed()</a></code> function. The inputs to the
<code><a href="../reference/get_seed.html">get_seed()</a></code> function in the siRNA sequence of interest and
the name of the seed.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the seed sequence information for the seed of interest</span></span>
<span><span class="va">seed</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_seed.html">get_seed</a></span><span class="op">(</span><span class="va">guide.seq</span>, <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seed</span></span>
<span><span class="co">#&gt; $Guide</span></span>
<span><span class="co">#&gt; 23-letter RNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFFFAF;">UU</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFFFAF;">UUUU</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Seed.Name</span></span>
<span><span class="co">#&gt; [1] "mer7m8"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Seed.Seq.RNA</span></span>
<span><span class="co">#&gt; 7-letter RNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Seed.Seq.DNA</span></span>
<span><span class="co">#&gt; 7-letter DNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Target.Seq</span></span>
<span><span class="co">#&gt; 7-letter DNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="counting-seed-matches-in-transcripts">Counting seed matches in transcripts<a class="anchor" aria-label="anchor" href="#counting-seed-matches-in-transcripts"></a>
</h3>
<p>You can perform a seed match for a single seed using the
<code><a href="../reference/SeedMatchR.html">SeedMatchR()</a></code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                 <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, </span>
<span>                 <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                 <span class="va">guide.seq</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0</span></span></code></pre></div>
<div class="section level4">
<h4 id="match-multiple-seeds">Match multiple seeds<a class="anchor" aria-label="anchor" href="#match-multiple-seeds"></a>
</h4>
<p>You can perform seed matching for all available seeds using a for
loop. The results will be appended as a new column to the results data
frame.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">seed</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mer8"</span>, <span class="st">"mer6"</span>, <span class="st">"mer7A1"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                  <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, </span>
<span>                  <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                  <span class="va">guide.seq</span>, </span>
<span>                  seed.name <span class="op">=</span> <span class="va">seed</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8 mer8 mer6 mer7A1</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1    1    1      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0    0    0      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1    0    1      0</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0    0    0      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0    0    0      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0    0    0      0</span></span></code></pre></div>
<div class="section level5">
<h5 id="match-seeds-with-mismatches-and-indels-allowed">Match seeds with mismatches and indels allowed<a class="anchor" aria-label="anchor" href="#match-seeds-with-mismatches-and-indels-allowed"></a>
</h5>
<p>You can also allow for inexact seed matches in your analysis with the
<code>mismatches</code> and <code>indels</code> arguments. The names can
be adjusted to reflect the arguments using the <code>col.name</code>
argument.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">indel.bool</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">mm</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">seed</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mer7m8"</span>, <span class="st">"mer8"</span>, <span class="st">"mer6"</span>, <span class="st">"mer7A1"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>      <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                        <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, </span>
<span>                        <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                        <span class="va">guide.seq</span>, </span>
<span>                        seed.name <span class="op">=</span> <span class="va">seed</span>, </span>
<span>                        col.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">seed</span>, <span class="st">"."</span>, <span class="st">"mm"</span>, <span class="va">mm</span>, <span class="st">"_indel"</span>, <span class="va">indel.bool</span><span class="op">)</span>, </span>
<span>                        mismatches <span class="op">=</span> <span class="va">mm</span>, </span>
<span>                        indels <span class="op">=</span> <span class="va">indel.bool</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8 mer8 mer6 mer7A1 mer7m8.mm0_indelTRUE</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1    1    1      1                    1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1    0    1      0                    1</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0    0    0      0                    0</span></span>
<span><span class="co">#&gt;   mer8.mm0_indelTRUE mer6.mm0_indelTRUE mer7A1.mm0_indelTRUE</span></span>
<span><span class="co">#&gt; 1                  1                  1                    1</span></span>
<span><span class="co">#&gt; 2                  0                  0                    0</span></span>
<span><span class="co">#&gt; 3                  0                  1                    0</span></span>
<span><span class="co">#&gt; 4                  0                  0                    0</span></span>
<span><span class="co">#&gt; 5                  0                  0                    0</span></span>
<span><span class="co">#&gt; 6                  0                  0                    0</span></span>
<span><span class="co">#&gt;   mer7m8.mm1_indelTRUE mer8.mm1_indelTRUE mer6.mm1_indelTRUE</span></span>
<span><span class="co">#&gt; 1                    1                  1                  2</span></span>
<span><span class="co">#&gt; 2                   11                  3                 29</span></span>
<span><span class="co">#&gt; 3                    3                  2                 12</span></span>
<span><span class="co">#&gt; 4                    0                  0                  6</span></span>
<span><span class="co">#&gt; 5                    1                  1                  1</span></span>
<span><span class="co">#&gt; 6                    1                  0                  7</span></span>
<span><span class="co">#&gt;   mer7A1.mm1_indelTRUE mer7m8.mm2_indelTRUE mer8.mm2_indelTRUE</span></span>
<span><span class="co">#&gt; 1                    1                    4                  2</span></span>
<span><span class="co">#&gt; 2                    7                   94                 21</span></span>
<span><span class="co">#&gt; 3                    6                   38                 15</span></span>
<span><span class="co">#&gt; 4                    4                   18                  6</span></span>
<span><span class="co">#&gt; 5                    1                    4                  2</span></span>
<span><span class="co">#&gt; 6                    1                   18                  6</span></span>
<span><span class="co">#&gt;   mer6.mm2_indelTRUE mer7A1.mm2_indelTRUE mer7m8.mm0_indelFALSE</span></span>
<span><span class="co">#&gt; 1                 14                    6                     1</span></span>
<span><span class="co">#&gt; 2                204                   58                     0</span></span>
<span><span class="co">#&gt; 3                101                   38                     1</span></span>
<span><span class="co">#&gt; 4                 41                   16                     0</span></span>
<span><span class="co">#&gt; 5                 14                    8                     0</span></span>
<span><span class="co">#&gt; 6                 50                   20                     0</span></span>
<span><span class="co">#&gt;   mer8.mm0_indelFALSE mer6.mm0_indelFALSE mer7A1.mm0_indelFALSE</span></span>
<span><span class="co">#&gt; 1                   1                   1                     1</span></span>
<span><span class="co">#&gt; 2                   0                   0                     0</span></span>
<span><span class="co">#&gt; 3                   0                   1                     0</span></span>
<span><span class="co">#&gt; 4                   0                   0                     0</span></span>
<span><span class="co">#&gt; 5                   0                   0                     0</span></span>
<span><span class="co">#&gt; 6                   0                   0                     0</span></span>
<span><span class="co">#&gt;   mer7m8.mm1_indelFALSE mer8.mm1_indelFALSE mer6.mm1_indelFALSE</span></span>
<span><span class="co">#&gt; 1                     1                   1                   1</span></span>
<span><span class="co">#&gt; 2                     6                   3                  18</span></span>
<span><span class="co">#&gt; 3                     3                   2                  12</span></span>
<span><span class="co">#&gt; 4                     0                   0                   2</span></span>
<span><span class="co">#&gt; 5                     1                   1                   1</span></span>
<span><span class="co">#&gt; 6                     1                   0                   6</span></span>
<span><span class="co">#&gt;   mer7A1.mm1_indelFALSE mer7m8.mm2_indelFALSE mer8.mm2_indelFALSE</span></span>
<span><span class="co">#&gt; 1                     1                     3                   1</span></span>
<span><span class="co">#&gt; 2                     6                    39                  12</span></span>
<span><span class="co">#&gt; 3                     6                    22                   8</span></span>
<span><span class="co">#&gt; 4                     1                     7                   1</span></span>
<span><span class="co">#&gt; 5                     1                     2                   1</span></span>
<span><span class="co">#&gt; 6                     0                     9                   2</span></span>
<span><span class="co">#&gt;   mer6.mm2_indelFALSE mer7A1.mm2_indelFALSE</span></span>
<span><span class="co">#&gt; 1                   6                     1</span></span>
<span><span class="co">#&gt; 2                 111                    35</span></span>
<span><span class="co">#&gt; 3                  62                    22</span></span>
<span><span class="co">#&gt; 4                  25                    10</span></span>
<span><span class="co">#&gt; 5                   7                     3</span></span>
<span><span class="co">#&gt; 6                  28                    11</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="comparing-the-expression-profiles-of-seed-targets-to-background">Comparing the expression profiles of seed targets to background<a class="anchor" aria-label="anchor" href="#comparing-the-expression-profiles-of-seed-targets-to-background"></a>
</h3>
<p>Many factors that perturb gene expression, like miRNA, show
cumulative changes in their targets gene expression. Cumulative changes
in the profile of genes expression can be visualized and tested with the
emperical distribution function (ecdf) coupled with a statistical test
such as the Kolmogorov-Smirnov test.</p>
<p><code>SeedMatchR</code> provides functions for comparing the
log2(Fold Change) of two gene sets. The function
<code>deseq_fc_ecdf</code> is designed to work directly with a DESeq2
results data frame.</p>
<p>Required Inputs:</p>
<ul>
<li>
<code>res</code>: DESeq2 results data frame</li>
<li>
<code>gene.lists</code>: A list of lists containing gene names</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene set 1 </span></span>
<span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8.mm0_indelFALSE</span> <span class="op">&gt;=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">res</span><span class="op">$</span><span class="va">mer8.mm0_indelFALSE</span> <span class="op">==</span><span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Gene set 2 </span></span>
<span><span class="va">mer8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer8.mm0_indelFALSE</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8.mm0_indelFALSE</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">&amp;</span> <span class="va">res</span><span class="op">$</span><span class="va">mer8.mm0_indelFALSE</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/deseq_fc_ecdf.html">deseq_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, <span class="st">"mer8"</span> <span class="op">=</span> <span class="va">mer8.list</span>, <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span><span class="op">)</span>,</span>
<span>                             stats.test <span class="op">=</span> <span class="st">"KS"</span>, </span>
<span>                             factor.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span>, <span class="st">"mer8"</span>, <span class="st">"mer7m8"</span><span class="op">)</span>, </span>
<span>                             null.name <span class="op">=</span> <span class="st">"Background"</span>,</span>
<span>                             target.name <span class="op">=</span> <span class="st">"mer8"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Comparing: Background vs. mer8"</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="SeedMatchR_files/figure-html/unnamed-chunk-16-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="using-seedmatchr-to-explore-potential-small-activating-rna-effects">Using <code>SeedMatchR</code> to explore potential small activating
RNA effects<a class="anchor" aria-label="anchor" href="#using-seedmatchr-to-explore-potential-small-activating-rna-effects"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Group transcripts by gene</span></span>
<span><span class="va">sequences</span> <span class="op">&lt;-</span> <span class="fu">transcriptsBy</span><span class="op">(</span><span class="va">anno.db</span><span class="op">$</span><span class="va">tx.db</span>, by<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract promoter sequences from tx.db object</span></span>
<span><span class="va">prom.seq</span> <span class="op">=</span> <span class="fu">getPromoterSeq</span><span class="op">(</span><span class="va">sequences</span>,</span>
<span>               <span class="va">anno.db</span><span class="op">$</span><span class="va">dna</span>,</span>
<span>                 upstream<span class="op">=</span><span class="fl">2000</span>,</span>
<span>                 downstream<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform a seed search of the promoter sequences. Set tx.id.col to F to use gene annotations</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, <span class="va">prom.seq</span><span class="op">@</span><span class="va">unlistData</span>, <span class="va">guide.seq</span>, tx.id.col <span class="op">=</span> <span class="cn">FALSE</span>, col.name <span class="op">=</span> <span class="st">"promoter.mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find the genes with matches</span></span>
<span><span class="va">promoterWseed</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">promoter.mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Generate the background list of genes</span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">promoterWseed</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Plot ecdf results for promoter matches with stats testing</span></span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/deseq_fc_ecdf.html">deseq_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             title <span class="op">=</span> <span class="st">"Ttr D1 30mkg"</span>,</span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, </span>
<span>                                  <span class="st">"Promoter w/ mer7m8"</span> <span class="op">=</span> <span class="va">promoterWseed</span><span class="op">)</span>,</span>
<span>                             stats.test <span class="op">=</span> <span class="st">"KS"</span>, </span>
<span>                             factor.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span>, </span>
<span>                                              <span class="st">"Promoter w/ mer7m8"</span><span class="op">)</span>, </span>
<span>                             null.name <span class="op">=</span> <span class="st">"Background"</span>,</span>
<span>                             target.name <span class="op">=</span> <span class="st">"Promoter w/ mer7m8"</span>,</span>
<span>                             palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"#d35400"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Comparing: Background vs. Promoter w/ mer7m8"</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="SeedMatchR_files/figure-html/unnamed-chunk-17-1.png" width="960"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] rtracklayer_1.60.0     GenomicFeatures_1.52.0 AnnotationDbi_1.62.1  </span></span>
<span><span class="co">#&gt;  [4] Biobase_2.60.0         GenomicRanges_1.52.0   msa_1.32.0            </span></span>
<span><span class="co">#&gt;  [7] Biostrings_2.68.1      GenomeInfoDb_1.36.0    XVector_0.40.0        </span></span>
<span><span class="co">#&gt; [10] IRanges_2.34.0         S4Vectors_0.38.1       BiocGenerics_0.46.0   </span></span>
<span><span class="co">#&gt; [13] SeedMatchR_1.0.0      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3            jsonlite_1.8.5               </span></span>
<span><span class="co">#&gt;   [3] magrittr_2.0.3                farver_2.1.1                 </span></span>
<span><span class="co">#&gt;   [5] rmarkdown_2.22                fs_1.6.2                     </span></span>
<span><span class="co">#&gt;   [7] BiocIO_1.10.0                 zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;   [9] ragg_1.2.5                    vctrs_0.6.2                  </span></span>
<span><span class="co">#&gt;  [11] memoise_2.0.1                 Rsamtools_2.16.0             </span></span>
<span><span class="co">#&gt;  [13] RCurl_1.98-1.12               ggtree_3.8.0                 </span></span>
<span><span class="co">#&gt;  [15] R4RNA_1.28.0                  htmltools_0.5.5              </span></span>
<span><span class="co">#&gt;  [17] S4Arrays_1.0.4                progress_1.2.2               </span></span>
<span><span class="co">#&gt;  [19] AnnotationHub_3.8.0           curl_5.0.1                   </span></span>
<span><span class="co">#&gt;  [21] gridGraphics_0.5-1            proj4_1.0-12                 </span></span>
<span><span class="co">#&gt;  [23] sass_0.4.6                    KernSmooth_2.23-21           </span></span>
<span><span class="co">#&gt;  [25] bslib_0.5.0                   desc_1.4.2                   </span></span>
<span><span class="co">#&gt;  [27] cachem_1.0.8                  GenomicAlignments_1.36.0     </span></span>
<span><span class="co">#&gt;  [29] ggalt_0.4.0                   mime_0.12                    </span></span>
<span><span class="co">#&gt;  [31] lifecycle_1.0.3               pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [33] Matrix_1.5-4.1                R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [35] fastmap_1.1.1                 GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt;  [37] MatrixGenerics_1.12.2         shiny_1.7.4                  </span></span>
<span><span class="co">#&gt;  [39] digest_0.6.31                 aplot_0.1.10                 </span></span>
<span><span class="co">#&gt;  [41] colorspace_2.1-0              patchwork_1.1.2              </span></span>
<span><span class="co">#&gt;  [43] rprojroot_2.0.3               textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [45] RSQLite_2.3.1                 labeling_0.4.2               </span></span>
<span><span class="co">#&gt;  [47] filelock_1.0.2                testit_0.13                  </span></span>
<span><span class="co">#&gt;  [49] fansi_1.0.4                   polyclip_1.10-4              </span></span>
<span><span class="co">#&gt;  [51] httr_1.4.6                    compiler_4.3.1               </span></span>
<span><span class="co">#&gt;  [53] bit64_4.0.5                   withr_2.5.0                  </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.34.2           DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [57] highr_0.10                    ggforce_0.4.1                </span></span>
<span><span class="co">#&gt;  [59] Rttf2pt1_1.3.12               maps_3.4.1                   </span></span>
<span><span class="co">#&gt;  [61] biomaRt_2.56.1                MASS_7.3-60                  </span></span>
<span><span class="co">#&gt;  [63] rappdirs_0.3.3                DelayedArray_0.26.3          </span></span>
<span><span class="co">#&gt;  [65] rjson_0.2.21                  tools_4.3.1                  </span></span>
<span><span class="co">#&gt;  [67] ape_5.7-1                     interactiveDisplayBase_1.38.0</span></span>
<span><span class="co">#&gt;  [69] httpuv_1.6.11                 extrafontdb_1.0              </span></span>
<span><span class="co">#&gt;  [71] glue_1.6.2                    restfulr_0.0.15              </span></span>
<span><span class="co">#&gt;  [73] nlme_3.1-162                  promises_1.2.0.1             </span></span>
<span><span class="co">#&gt;  [75] grid_4.3.1                    generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [77] gtable_0.3.3                  tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt;  [79] hms_1.1.3                     xml2_1.3.4                   </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.3                    BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt;  [83] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">#&gt;  [85] yulab.utils_0.0.6             later_1.3.1                  </span></span>
<span><span class="co">#&gt;  [87] tweenr_2.0.2                  dplyr_1.1.2                  </span></span>
<span><span class="co">#&gt;  [89] treeio_1.24.1                 BiocFileCache_2.8.0          </span></span>
<span><span class="co">#&gt;  [91] lattice_0.21-8                bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [93] tidyselect_1.2.0              knitr_1.43                   </span></span>
<span><span class="co">#&gt;  [95] SummarizedExperiment_1.30.2   xfun_0.39                    </span></span>
<span><span class="co">#&gt;  [97] matrixStats_1.0.0             seqmagick_0.1.5              </span></span>
<span><span class="co">#&gt;  [99] stringi_1.7.12                lazyeval_0.2.2               </span></span>
<span><span class="co">#&gt; [101] ggfun_0.0.9                   yaml_2.3.7                   </span></span>
<span><span class="co">#&gt; [103] evaluate_0.21                 codetools_0.2-19             </span></span>
<span><span class="co">#&gt; [105] extrafont_0.19                twosamples_2.0.0             </span></span>
<span><span class="co">#&gt; [107] tibble_3.2.1                  ggmsa_1.6.0                  </span></span>
<span><span class="co">#&gt; [109] BiocManager_1.30.21           ggplotify_0.1.0              </span></span>
<span><span class="co">#&gt; [111] cli_3.6.1                     ash_1.0-15                   </span></span>
<span><span class="co">#&gt; [113] xtable_1.8-4                  systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [115] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [117] Rcpp_1.0.10                   dbplyr_2.3.2                 </span></span>
<span><span class="co">#&gt; [119] png_0.1-8                     XML_3.99-0.14                </span></span>
<span><span class="co">#&gt; [121] parallel_4.3.1                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [123] pkgdown_2.0.7                 ggplot2_3.4.2                </span></span>
<span><span class="co">#&gt; [125] blob_1.2.4                    prettyunits_1.1.1            </span></span>
<span><span class="co">#&gt; [127] bitops_1.0-7                  tidytree_0.4.2               </span></span>
<span><span class="co">#&gt; [129] scales_1.2.1                  purrr_1.0.1                  </span></span>
<span><span class="co">#&gt; [131] crayon_1.5.2                  rlang_1.1.1                  </span></span>
<span><span class="co">#&gt; [133] cowplot_1.1.1                 KEGGREST_1.40.0</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tareian Cazares.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
