<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find Matches to siRNA in Genomic Features with SeedMatchR • SeedMatchR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Matches to siRNA in Genomic Features with SeedMatchR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find Matches to siRNA in Genomic Features with SeedMatchR</h1>
                        <h4 data-toc-skip class="author">Tareian
Cazares, Pradeep Pugalenthi, Richard Higgs, Jibo Wang, and Hatice Gulcin
Ozer</h4>
            
      

      <div class="d-none name"><code>SeedMatchR.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>On-target siRNA silencing of mRNA results from binding to fully
      complementary regions in transcripts to induce cleavage.
      Off-target siRNA gene knockdown can occur through several modes,
      one being a seed-mediated mechanism mimicking miRNA gene
      regulation. Seed-mediated off-target effects occur when the ~8
      nucleotides at the 5’ end of the guide strand, called a seed
      region, bind the 3’ untranslated regions of mRNA, causing reduced
      translation. Experiments using siRNA knockdown paired with RNA-seq
      can be used to detect siRNA sequences with potential off-target
      effects driven by the seed region. ‘SeedMatchR’ provides tools for
      exploring and detecting potential seed-mediated off-target effects
      of siRNA in RNA-seq experiments. ‘SeedMatchR’ is designed to
      extend current differential expression analysis tools, such as
      ‘DESeq2’, by annotating results with predicted seed matches. Using
      publicly available data, we demonstrate the ability of
      ‘SeedMatchR’ to detect cumulative changes in differential gene
      expression attributed to siRNA seed regions.</p>
    </div>
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of SeedMatchR from <a href="https://github.com/" class="external-link">GitHub</a> or the stable build from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tacazares/SeedMatchR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start-example-with-public-sirna-data">Quick start example with public siRNA data<a class="anchor" aria-label="anchor" href="#quick-start-example-with-public-sirna-data"></a>
</h2>
<div class="section level3">
<h3 id="introduction-to-example-dataset">Introduction to example dataset<a class="anchor" aria-label="anchor" href="#introduction-to-example-dataset"></a>
</h3>
<p>This example uses the siRNA sequence, D1, targeting the Ttr gene in
rat liver from the publication:</p>
<pre><code>Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga-Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp-Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside: Improving the clinical safety of GalNAc-siRNA conjugates using seed-pairing destabilization. Nucleic Acids Res. 2022 Jul 8;50(12):6656-6670. doi: 10.1093/nar/gkac539. PMID: 35736224; PMCID: PMC9262600.</code></pre>
<p>The guide sequence of interest is 23 bp long and oriented 5’ -&gt;
3’.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># siRNA sequence of interest targeting a 23 bp region of the Ttr gene</span></span>
<span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-modes-and-input-data">Run modes and input data<a class="anchor" aria-label="anchor" href="#run-modes-and-input-data"></a>
</h3>
<p><code>SeedMatchR</code> can be run in two different modes. The first
mode is designed for <code>DESEQ2</code> differential expression results
and the second mode is designed for finding exact match positions
regardless of differential expression data.</p>
<ul>
<li>
<code>DESEQ2</code> mode will map a column of match counts to a
<code>data.frame</code> of differential expression resutls.</li>
<li>
<code>data.frame</code> model will return a <code>data.frame</code>
of matches that correspond to the interval ranges for matches in the
input <code>DNAStringSet</code> library being searched.</li>
</ul>
<p>The required inputs to <code>SeedMatchR</code> are list below:</p>
<ul>
<li>A character string representing the siRNA RNA sequence. This must be
greater than 8 bp.</li>
<li>
<code>GTF</code>: gene transfer file containing species specific
genomic information for gene bodies. This is used to derive the list of
3’ UTRs and other features used in the analysis. This is also used to
map transcript IDs to gene IDs.</li>
<li>Feature-specific <code>DNAStringSet</code>: A
<code>DNAStringSet</code> object of sequences for each of the features
of interest. The features must be named according to the transcript they
were derived from. Examples include those generated by
<code><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html" class="external-link">GenomicFeatures::extractTranscriptSeqs()</a></code> paired with
functions like
<code><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html" class="external-link">GenomicFeatures::threeUTRsByTranscript()</a></code>.</li>
<li>Seed name of the canonical seed you want to search or a custom
definition with associated start and stop positions on the siRNA.</li>
</ul>
<p>Optional input:</p>
<ul>
<li>
<code>res</code>: A data frame of DESEQ2 results. If a DESEQ2
results are provided as a <code>data.frame</code>, then
<code>SeedMatchR</code> will map that total match counts as a new column
in the <code>data.frame</code>.</li>
</ul>
<p><code>SeedMatchR</code> provides functions for helping to wrangle the
many data types and annotations. <code>SeedMatchR</code> makes extensive
use of <code>AnnotationHub</code> to help access genomic annotations in
a reproducible manner.</p>
<div class="section level4">
<h4 id="prepare-species-specific-annotation-data">Prepare species-specific annotation data<a class="anchor" aria-label="anchor" href="#prepare-species-specific-annotation-data"></a>
</h4>
<p>The function <code><a href="../reference/load_annotations.html">load_annotations()</a></code> has built in annotation
data for human, rat, and mouse annotations. More information can be
found in <a href="articles/SeedMatchR_Annotation_Databases.Rmd">the
vignette on annotations</a>. We can load the species specific
annotations using the approach described below. Running this function
will take several minutes. Therefore it might be helpful to save the
objects and reload them later if you plan to use this code in a repeated
workflow.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the species specific annotation database object</span></span>
<span><span class="va">annodb</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_annotations.html">load_annotations</a></span><span class="op">(</span>reference.name <span class="op">=</span> <span class="st">"rnor6"</span>, canonical <span class="op">=</span> <span class="cn">FALSE</span>, min.feature.width <span class="op">=</span> <span class="fl">8</span>, longest.utr <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="prepare-deseq2-results">Prepare DESEQ2 Results<a class="anchor" aria-label="anchor" href="#prepare-deseq2-results"></a>
</h4>
<p>The <code>DESEQ2</code> mode of <code>SeedMatchR</code> assumes that
you will be performing your analysis on <code>DESEQ2</code> results
outputs. The first step is to load your <code>DESEQ2</code> results file
as a <code>data.frame</code>.</p>
<p>The test data that is provided with <code>SeedMatchR</code> was
derived from the 2022 publication by Schlegel et al. The data set
represents a DESeq2 analysis performed on rat liver that had been
treated with Ttr targeting siRNA. We will use this example to explore
seed mediated activity. The data set name is long, so it will be renamed
to <code>res</code>.</p>
</div>
<div class="section level4">
<h4 id="download-data-only-need-to-perform-once">Download data (only need to perform once)<a class="anchor" aria-label="anchor" href="#download-data-only-need-to-perform-once"></a>
</h4>
<p>We start by downloading the example data set. This function will
download three files from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929" class="external-link">GSE184929</a>.
These files represent three samples with different siRNA treatments at
two dosages.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Example data directory being created at: /home/runner/.local/share/R/SeedMatchR</span></span>
<span><span class="co">#&gt; Warning in dir.create(data.path, recursive = TRUE):</span></span>
<span><span class="co">#&gt; '/home/runner/.local/share/R/SeedMatchR' already exists</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h4>
<p>We can load the example data into the environment.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sirna.data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
<p>The DESeq2 results are available through the names
<code>Schlegel_2022_Ttr_D1_30mkg</code>,
<code>Schlegel_2022_Ttr_D4_30mkg</code> and
<code>Schlegel_2022_Ttr_D1_10mkg</code>. The data set name is long, so
it will be renamed to <code>res</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">sirna.data</span><span class="op">$</span><span class="va">Schlegel_2022_Ttr_D1_30mkg</span></span></code></pre></div>
<p>The <code>DESEQ2</code> results file is then filtered. The function
<code><a href="../reference/filter_res.html">filter_res()</a></code> can be used to filter a results file by
log2FoldChange, padj, baseMean, and remove NA entries.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dimensions before filtering</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 32883    6</span></span>
<span><span class="co">#&gt; [1] 32883     8</span></span>
<span></span>
<span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_res.html">filter_res</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Dimensions after filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="co"># [1] 13582     8</span></span>
<span><span class="co">#&gt; [1] 8124    8</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="get-the-seed-sequence-of-interest">Get the seed sequence of interest<a class="anchor" aria-label="anchor" href="#get-the-seed-sequence-of-interest"></a>
</h3>
<p>You can extract the seed sequence information from the siRNA input
sequence using the <code><a href="../reference/get_seed.html">get_seed()</a></code> function. The inputs to the
<code><a href="../reference/get_seed.html">get_seed()</a></code> function are the siRNA sequence of interest and
the name of the seed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the seed sequence information for the seed of interest</span></span>
<span><span class="va">seed</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_seed.html">get_seed</a></span><span class="op">(</span>guide.seq <span class="op">=</span> <span class="va">guide.seq</span>, seed.name <span class="op">=</span> <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seed</span></span>
<span><span class="co">#&gt; $Seed.Name</span></span>
<span><span class="co">#&gt; [1] "mer7m8"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Guide</span></span>
<span><span class="co">#&gt; 23-letter RNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFFFAF;">UU</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFFFAF;">UUUU</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Seed.Seq.RNA</span></span>
<span><span class="co">#&gt; 7-letter RNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFFFAF;">U</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Seed.Seq.DNA</span></span>
<span><span class="co">#&gt; 7-letter DNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Target.Seq</span></span>
<span><span class="co">#&gt; 7-letter DNAString object</span></span>
<span><span class="co">#&gt; seq: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="counting-seed-matches-in-transcripts">Counting seed matches in transcripts<a class="anchor" aria-label="anchor" href="#counting-seed-matches-in-transcripts"></a>
</h3>
<p>You can perform a seed match for a single seed using the
<code><a href="../reference/SeedMatchR.html">SeedMatchR()</a></code> function.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, </span>
<span>                 seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                 sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                 seed.name <span class="op">=</span> <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0</span></span></code></pre></div>
<div class="section level4">
<h4 id="match-multiple-seeds">Match multiple seeds<a class="anchor" aria-label="anchor" href="#match-multiple-seeds"></a>
</h4>
<p>You can perform seed matching for all available seeds using a for
loop. The results will be appended as a new column to the results data
frame.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">seed</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mer8"</span>, <span class="st">"mer6"</span>, <span class="st">"mer7A1"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, </span>
<span>                 seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                 sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                 seed.name <span class="op">=</span> <span class="va">seed</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8 mer8 mer6 mer7A1</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1    1    1      1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0    0    0      0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1    0    1      0</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0    0    0      0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0    0    0      0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0    0    0      0</span></span></code></pre></div>
<div class="section level5">
<h5 id="match-seeds-with-mismatches-and-indels-allowed">Match seeds with mismatches and indels allowed<a class="anchor" aria-label="anchor" href="#match-seeds-with-mismatches-and-indels-allowed"></a>
</h5>
<p>You can also allow for inexact seed matches in your analysis with the
<code>mismatches</code> and <code>indels</code> arguments. The names can
be adjusted to reflect the arguments using the <code>col.name</code>
argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="kw">for</span> <span class="op">(</span><span class="va">indel.bool</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">mm</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>      <span class="kw">for</span> <span class="op">(</span><span class="va">seed</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mer7m8"</span>, <span class="st">"mer8"</span>, <span class="st">"mer6"</span>, <span class="st">"mer7A1"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, </span>
<span>                        seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                        sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                        seed.name <span class="op">=</span> <span class="va">seed</span>,</span>
<span>                        col.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">seed</span>, </span>
<span>                                          <span class="st">"."</span>, </span>
<span>                                          <span class="st">"mm"</span>, <span class="va">mm</span>, </span>
<span>                                          <span class="st">"_indel"</span>, <span class="va">indel.bool</span><span class="op">)</span>, </span>
<span>                        max.mismatch <span class="op">=</span> <span class="va">mm</span>, </span>
<span>                        with.indels <span class="op">=</span> <span class="va">indel.bool</span><span class="op">)</span></span>
<span>      <span class="op">}</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id  baseMean log2FoldChange     lfcSE      stat        pvalue</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 2138.0945      -8.164615        NA -23.61818 2.507268e-123</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000127  437.6342      -1.346927 0.1068629 -12.60425  2.000712e-36</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000047179 1590.1745      -1.262411 0.1031403 -12.23974  1.906387e-34</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000030187  131.9206       3.422725 0.3032352  11.28736  1.515189e-29</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000008050   38.9921      -3.442834 0.3192776 -10.78320  4.132589e-27</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000008816  400.9526       2.794453 0.2661369  10.50006  8.632549e-26</span></span>
<span><span class="co">#&gt;            padj symbol mer7m8 mer8 mer6 mer7A1 mer7m8.mm0_indelTRUE</span></span>
<span><span class="co">#&gt; 1 3.405371e-119    Ttr      1    1    1      1                    1</span></span>
<span><span class="co">#&gt; 2  1.358683e-32  Kpna6      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 3  8.630849e-31  Aplp2      1    0    1      0                    1</span></span>
<span><span class="co">#&gt; 4  5.144824e-26  Mmp12      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 5  1.122577e-23  Stac3      0    0    0      0                    0</span></span>
<span><span class="co">#&gt; 6  1.954121e-22  Gpnmb      0    0    0      0                    0</span></span>
<span><span class="co">#&gt;   mer8.mm0_indelTRUE mer6.mm0_indelTRUE mer7A1.mm0_indelTRUE</span></span>
<span><span class="co">#&gt; 1                  1                  1                    1</span></span>
<span><span class="co">#&gt; 2                  0                  0                    0</span></span>
<span><span class="co">#&gt; 3                  0                  1                    0</span></span>
<span><span class="co">#&gt; 4                  0                  0                    0</span></span>
<span><span class="co">#&gt; 5                  0                  0                    0</span></span>
<span><span class="co">#&gt; 6                  0                  0                    0</span></span>
<span><span class="co">#&gt;   mer7m8.mm1_indelTRUE mer8.mm1_indelTRUE mer6.mm1_indelTRUE</span></span>
<span><span class="co">#&gt; 1                    1                  1                  2</span></span>
<span><span class="co">#&gt; 2                   11                  3                 29</span></span>
<span><span class="co">#&gt; 3                    3                  2                 12</span></span>
<span><span class="co">#&gt; 4                    0                  0                  6</span></span>
<span><span class="co">#&gt; 5                    1                  1                  1</span></span>
<span><span class="co">#&gt; 6                    1                  0                  7</span></span>
<span><span class="co">#&gt;   mer7A1.mm1_indelTRUE mer7m8.mm0_indelFALSE mer8.mm0_indelFALSE</span></span>
<span><span class="co">#&gt; 1                    1                     1                   1</span></span>
<span><span class="co">#&gt; 2                    7                     0                   0</span></span>
<span><span class="co">#&gt; 3                    6                     1                   0</span></span>
<span><span class="co">#&gt; 4                    4                     0                   0</span></span>
<span><span class="co">#&gt; 5                    1                     0                   0</span></span>
<span><span class="co">#&gt; 6                    1                     0                   0</span></span>
<span><span class="co">#&gt;   mer6.mm0_indelFALSE mer7A1.mm0_indelFALSE mer7m8.mm1_indelFALSE</span></span>
<span><span class="co">#&gt; 1                   1                     1                     1</span></span>
<span><span class="co">#&gt; 2                   0                     0                     6</span></span>
<span><span class="co">#&gt; 3                   1                     0                     3</span></span>
<span><span class="co">#&gt; 4                   0                     0                     0</span></span>
<span><span class="co">#&gt; 5                   0                     0                     1</span></span>
<span><span class="co">#&gt; 6                   0                     0                     1</span></span>
<span><span class="co">#&gt;   mer8.mm1_indelFALSE mer6.mm1_indelFALSE mer7A1.mm1_indelFALSE</span></span>
<span><span class="co">#&gt; 1                   1                   1                     1</span></span>
<span><span class="co">#&gt; 2                   3                  18                     6</span></span>
<span><span class="co">#&gt; 3                   2                  12                     6</span></span>
<span><span class="co">#&gt; 4                   0                   2                     1</span></span>
<span><span class="co">#&gt; 5                   1                   1                     1</span></span>
<span><span class="co">#&gt; 6                   0                   6                     0</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="returning-match-positions">Returning match positions<a class="anchor" aria-label="anchor" href="#returning-match-positions"></a>
</h3>
<div class="section level4">
<h4 id="return-as-data-frame">Return as data frame<a class="anchor" aria-label="anchor" href="#return-as-data-frame"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">match.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                      sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                      seed.name <span class="op">=</span> <span class="st">"mer7m8"</span>, </span>
<span>                      res.format <span class="op">=</span> <span class="st">"data.frame"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">match.df</span><span class="op">)</span></span>
<span><span class="co">#&gt;             seqnames start  end width</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000000091  2427 2433     7</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000170  3031 3037     7</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000000299   217  223     7</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000000415  1198 1204     7</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000000543   659  665     7</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000000569   777  783     7</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="return-as-granges">Return as GRanges<a class="anchor" aria-label="anchor" href="#return-as-granges"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">match.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                      sequence <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                      seed.name <span class="op">=</span> <span class="st">"mer7m8"</span>, </span>
<span>                      res.format <span class="op">=</span> <span class="st">"granges"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">match.df</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;                 seqnames    ranges strand</span></span>
<span><span class="co">#&gt;                    &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1] ENSRNOG00000000091 2427-2433      *</span></span>
<span><span class="co">#&gt;   [2] ENSRNOG00000000170 3031-3037      *</span></span>
<span><span class="co">#&gt;   [3] ENSRNOG00000000299   217-223      *</span></span>
<span><span class="co">#&gt;   [4] ENSRNOG00000000415 1198-1204      *</span></span>
<span><span class="co">#&gt;   [5] ENSRNOG00000000543   659-665      *</span></span>
<span><span class="co">#&gt;   [6] ENSRNOG00000000569   777-783      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 541 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="scan-all-sequences-for-all-match-types">Scan all sequences for all match types<a class="anchor" aria-label="anchor" href="#scan-all-sequences-for-all-match-types"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">match.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/full_search.html">full_search</a></span><span class="op">(</span>guide.seq <span class="op">=</span> <span class="va">guide.seq</span>, </span>
<span>                       seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                       <span class="st">"Ttr"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">match.df</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 seqs Full.mm0 Full.mm1 Full.mm2 Full.mm3 Full.mm4 18mer.mm0</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000000007        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000000008        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000000009        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000000010        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000000012        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000000017        0        0        0        0        0         0</span></span>
<span><span class="co">#&gt;   18mer.mm1 18mer.mm2 18mer.mm3 18mer.mm4 15mer.mm0 15mer.mm1 15mer.mm2</span></span>
<span><span class="co">#&gt; 1         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; 2         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; 3         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; 4         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; 5         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; 6         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt;   15mer.mm3 15mer.mm4 mer8.mm0 mer8.mm1 mer7m8.mm0 mer7m8.mm1 mer7A1.mm0</span></span>
<span><span class="co">#&gt; 1         0         0        0        0          0          2          0</span></span>
<span><span class="co">#&gt; 2         0         0        0        0          0          0          0</span></span>
<span><span class="co">#&gt; 3         0         0        0        0          0          0          0</span></span>
<span><span class="co">#&gt; 4         1         1        0        1          0          1          0</span></span>
<span><span class="co">#&gt; 5         0         0        0        0          0          0          0</span></span>
<span><span class="co">#&gt; 6         0         0        0        0          0          0          0</span></span>
<span><span class="co">#&gt;   mer7A1.mm1 mer6.mm0 mer6.mm1 Group best.match</span></span>
<span><span class="co">#&gt; 1          0        0       16   Ttr No.matches</span></span>
<span><span class="co">#&gt; 2          0        0        0   Ttr No.matches</span></span>
<span><span class="co">#&gt; 3          0        0        3   Ttr No.matches</span></span>
<span><span class="co">#&gt; 4          1        0        2   Ttr   15mer-D3</span></span>
<span><span class="co">#&gt; 5          0        0        0   Ttr No.matches</span></span>
<span><span class="co">#&gt; 6          0        0        1   Ttr No.matches</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-a-seedmatchreport">Create a SeedMatchReport<a class="anchor" aria-label="anchor" href="#create-a-seedmatchreport"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">report</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchReport.html">SeedMatchReport</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">res</span>, </span>
<span>                         seqs <span class="op">=</span> <span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span>, </span>
<span>                         group.name <span class="op">=</span> <span class="st">"Ttr"</span>, </span>
<span>                         guide.seq <span class="op">=</span> <span class="va">guide.seq</span><span class="op">)</span></span>
<span></span>
<span><span class="va">report</span><span class="op">$</span><span class="va">table</span></span></code></pre></div>
<div id="mygt" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#mygt table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#mygt thead, #mygt tbody, #mygt tfoot, #mygt tr, #mygt td, #mygt th {
  border-style: none;
}

#mygt p {
  margin: 0;
  padding: 0;
}

#mygt .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #929292;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#mygt .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#mygt .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#mygt .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#mygt .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mygt .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
}

#mygt .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #929292;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mygt .gt_col_heading {
  color: #FFFFFF;
  background-color: #004D80;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#mygt .gt_column_spanner_outer {
  color: #FFFFFF;
  background-color: #004D80;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#mygt .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#mygt .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#mygt .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#mygt .gt_spanner_row {
  border-bottom-style: hidden;
}

#mygt .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #929292;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#mygt .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #929292;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
  vertical-align: middle;
}

#mygt .gt_from_md > :first-child {
  margin-top: 0;
}

#mygt .gt_from_md > :last-child {
  margin-bottom: 0;
}

#mygt .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: dashed;
  border-top-width: 1px;
  border-top-color: #929292;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #929292;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #929292;
  vertical-align: middle;
  overflow-x: hidden;
}

#mygt .gt_stub {
  color: #333333;
  background-color: #D5D5D5;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #FFFFFF;
  padding-left: 5px;
  padding-right: 5px;
}

#mygt .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#mygt .gt_row_group_first td {
  border-top-width: 2px;
}

#mygt .gt_row_group_first th {
  border-top-width: 2px;
}

#mygt .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mygt .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #929292;
}

#mygt .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#mygt .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
}

#mygt .gt_grand_summary_row {
  color: #333333;
  background-color: #D5D5D5;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mygt .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #929292;
}

#mygt .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #929292;
}

#mygt .gt_striped {
  background-color: #F4F4F4;
}

#mygt .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #929292;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #929292;
}

#mygt .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mygt .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mygt .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mygt .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mygt .gt_left {
  text-align: left;
}

#mygt .gt_center {
  text-align: center;
}

#mygt .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#mygt .gt_font_normal {
  font-weight: normal;
}

#mygt .gt_font_bold {
  font-weight: bold;
}

#mygt .gt_font_italic {
  font-style: italic;
}

#mygt .gt_super {
  font-size: 65%;
}

#mygt .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#mygt .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#mygt .gt_indent_1 {
  text-indent: 5px;
}

#mygt .gt_indent_2 {
  text-indent: 10px;
}

#mygt .gt_indent_3 {
  text-indent: 15px;
}

#mygt .gt_indent_4 {
  text-indent: 20px;
}

#mygt .gt_indent_5 {
  text-indent: 25px;
}

#mygt .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#mygt div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading">
<td colspan="21" class="gt_heading gt_title gt_font_normal" style>In-silico siRNA Binding Prediction</td>
    </tr>
<tr class="gt_heading">
<td colspan="21" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border" style>Identifying siRNA hits in the transcriptome</td>
    </tr>
<tr class="gt_col_headings gt_spanner_row">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="2" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="5" scope="colgroup" id="Full Guide Strand (g2:g23)">
        <div class="gt_column_spanner">Full Guide Strand (g2:g23)</div>
      </th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="5" scope="colgroup" id="18-mer (g2:g19)">
        <div class="gt_column_spanner">18-mer (g2:g19)</div>
      </th>
      <th class="gt_center gt_columns_top_border gt_column_spanner_outer" rowspan="1" colspan="5" scope="colgroup" id="15-mer (g2:g19)">
        <div class="gt_column_spanner">15-mer (g2:g19)</div>
      </th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="2" colspan="1" scope="col" id="a8mer">8mer</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="2" colspan="1" scope="col" id="a7mer-m8">7mer-m8</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="2" colspan="1" scope="col" id="a7mer-A1">7mer-A1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="2" colspan="1" scope="col" id="a6mer">6mer</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="2" colspan="1" scope="col" id="Total">Total</th>
    </tr>
<tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Full-D0">D0</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Full-D1">D1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Full-D2">D2</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Full-D3">D3</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Full-D4">D4</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a18mer-D0">D0</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a18mer-D1">D1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a18mer-D2">D2</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a18mer-D3">D3</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a18mer-D4">D4</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a15mer-D0">D0</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a15mer-D1">D1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a15mer-D2">D2</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a15mer-D3">D3</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="a15mer-D4">D4</th>
    </tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row">
<th colspan="21" class="gt_group_heading" scope="colgroup" id="Ttr">Ttr</th>
    </tr>
<tr class="gt_row_group_first">
<th id="stub_1_1" scope="row" class="gt_row gt_left gt_stub">In silico predictions</th>
<td headers="Ttr stub_1_1 Full-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 Full-D1" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_1 Full-D2" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 Full-D3" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_1 Full-D4" class="gt_row gt_right">20</td>
<td headers="Ttr stub_1_1 18mer-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 18mer-D1" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 18mer-D2" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_1 18mer-D3" class="gt_row gt_right">82</td>
<td headers="Ttr stub_1_1 18mer-D4" class="gt_row gt_right">1,124</td>
<td headers="Ttr stub_1_1 15mer-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 15mer-D1" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_1 15mer-D2" class="gt_row gt_right">8</td>
<td headers="Ttr stub_1_1 15mer-D3" class="gt_row gt_right">1,100</td>
<td headers="Ttr stub_1_1 15mer-D4" class="gt_row gt_right">6,765</td>
<td headers="Ttr stub_1_1 8mer" class="gt_row gt_right">15</td>
<td headers="Ttr stub_1_1 7mer-m8" class="gt_row gt_right">27</td>
<td headers="Ttr stub_1_1 7mer-A1" class="gt_row gt_right">55</td>
<td headers="Ttr stub_1_1 6mer" class="gt_row gt_right">135</td>
<td headers="Ttr stub_1_1 Total" class="gt_row gt_right">9,334</td>
</tr>
<tr>
<th id="stub_1_2" scope="row" class="gt_row gt_left gt_stub">Expressed predictions</th>
<td headers="Ttr stub_1_2 Full-D0" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 Full-D1" class="gt_row gt_right gt_striped">1</td>
<td headers="Ttr stub_1_2 Full-D2" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 Full-D3" class="gt_row gt_right gt_striped">1</td>
<td headers="Ttr stub_1_2 Full-D4" class="gt_row gt_right gt_striped">11</td>
<td headers="Ttr stub_1_2 18mer-D0" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 18mer-D1" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 18mer-D2" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 18mer-D3" class="gt_row gt_right gt_striped">53</td>
<td headers="Ttr stub_1_2 18mer-D4" class="gt_row gt_right gt_striped">663</td>
<td headers="Ttr stub_1_2 15mer-D0" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 15mer-D1" class="gt_row gt_right gt_striped">0</td>
<td headers="Ttr stub_1_2 15mer-D2" class="gt_row gt_right gt_striped">3</td>
<td headers="Ttr stub_1_2 15mer-D3" class="gt_row gt_right gt_striped">642</td>
<td headers="Ttr stub_1_2 15mer-D4" class="gt_row gt_right gt_striped">3,483</td>
<td headers="Ttr stub_1_2 8mer" class="gt_row gt_right gt_striped">6</td>
<td headers="Ttr stub_1_2 7mer-m8" class="gt_row gt_right gt_striped">9</td>
<td headers="Ttr stub_1_2 7mer-A1" class="gt_row gt_right gt_striped">31</td>
<td headers="Ttr stub_1_2 6mer" class="gt_row gt_right gt_striped">66</td>
<td headers="Ttr stub_1_2 Total" class="gt_row gt_right gt_striped">4,969</td>
</tr>
<tr>
<th id="stub_1_3" scope="row" class="gt_row gt_left gt_stub">Off-target predictions</th>
<td headers="Ttr stub_1_3 Full-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 Full-D1" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_3 Full-D2" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 Full-D3" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_3 Full-D4" class="gt_row gt_right">2</td>
<td headers="Ttr stub_1_3 18mer-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 18mer-D1" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 18mer-D2" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 18mer-D3" class="gt_row gt_right">5</td>
<td headers="Ttr stub_1_3 18mer-D4" class="gt_row gt_right">31</td>
<td headers="Ttr stub_1_3 15mer-D0" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 15mer-D1" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 15mer-D2" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 15mer-D3" class="gt_row gt_right">28</td>
<td headers="Ttr stub_1_3 15mer-D4" class="gt_row gt_right">124</td>
<td headers="Ttr stub_1_3 8mer" class="gt_row gt_right">1</td>
<td headers="Ttr stub_1_3 7mer-m8" class="gt_row gt_right">0</td>
<td headers="Ttr stub_1_3 7mer-A1" class="gt_row gt_right">2</td>
<td headers="Ttr stub_1_3 6mer" class="gt_row gt_right">7</td>
<td headers="Ttr stub_1_3 Total" class="gt_row gt_right">202</td>
</tr>
<tr>
<th id="stub_1_4" scope="row" class="gt_row gt_left gt_stub">% off-target</th>
<td headers="Ttr stub_1_4 Full-D0" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 Full-D1" class="gt_row gt_right gt_striped" style="background-color: #FF0000; color: #FFFFFF;">100.00%</td>
<td headers="Ttr stub_1_4 Full-D2" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 Full-D3" class="gt_row gt_right gt_striped" style="background-color: #FF0000; color: #FFFFFF;">100.00%</td>
<td headers="Ttr stub_1_4 Full-D4" class="gt_row gt_right gt_striped" style="background-color: #FFDCD0; color: #000000;">18.18%</td>
<td headers="Ttr stub_1_4 18mer-D0" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 18mer-D1" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 18mer-D2" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 18mer-D3" class="gt_row gt_right gt_striped" style="background-color: #FFEDE6; color: #000000;">9.43%</td>
<td headers="Ttr stub_1_4 18mer-D4" class="gt_row gt_right gt_striped" style="background-color: #FFF6F3; color: #000000;">4.68%</td>
<td headers="Ttr stub_1_4 15mer-D0" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 15mer-D1" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 15mer-D2" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 15mer-D3" class="gt_row gt_right gt_striped" style="background-color: #FFF7F4; color: #000000;">4.36%</td>
<td headers="Ttr stub_1_4 15mer-D4" class="gt_row gt_right gt_striped" style="background-color: #FFF8F6; color: #000000;">3.56%</td>
<td headers="Ttr stub_1_4 8mer" class="gt_row gt_right gt_striped" style="background-color: #FFDFD4; color: #000000;">16.67%</td>
<td headers="Ttr stub_1_4 7mer-m8" class="gt_row gt_right gt_striped" style="background-color: #FFFFFF; color: #000000;">0.00%</td>
<td headers="Ttr stub_1_4 7mer-A1" class="gt_row gt_right gt_striped" style="background-color: #FFF3EE; color: #000000;">6.45%</td>
<td headers="Ttr stub_1_4 6mer" class="gt_row gt_right gt_striped" style="background-color: #FFEBE3; color: #000000;">10.61%</td>
<td headers="Ttr stub_1_4 Total" class="gt_row gt_right gt_striped">4.07%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="comparing-the-expression-profiles-of-seed-targets-to-background">Comparing the expression profiles of seed targets to background<a class="anchor" aria-label="anchor" href="#comparing-the-expression-profiles-of-seed-targets-to-background"></a>
</h3>
<p>Many factors that perturb gene expression, like miRNA, show
cumulative changes in their targets gene expression. Cumulative changes
in the profile of genes expression can be visualized and tested with the
emperical distribution function (ecdf) coupled with a statistical test
such as the Kolmogorov-Smirnov test.</p>
<p><code>SeedMatchR</code> provides functions for comparing the
log2(Fold Change) of two gene sets. The function
<code>deseq_fc_ecdf</code> is designed to work directly with a DESeq2
results data frame.</p>
<p>Required Inputs:</p>
<ul>
<li>
<code>res</code>: <code>DESEQ2</code> results
<code>data.frame</code>
</li>
<li>
<code>gene.lists</code>: A list of lists containing gene names</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene set 1 </span></span>
<span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">res</span><span class="op">$</span><span class="va">mer8</span> <span class="op">==</span><span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Gene set 2 </span></span>
<span><span class="va">mer8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer8</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">&amp;</span> <span class="va">res</span><span class="op">$</span><span class="va">mer8</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/deseq_fc_ecdf.html">deseq_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             title <span class="op">=</span> <span class="st">"ECDF of log2(FC)"</span>,</span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, </span>
<span>                                  <span class="st">"mer8"</span> <span class="op">=</span> <span class="va">mer8.list</span>, <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span><span class="op">)</span>,</span>
<span>                             stats_test <span class="op">=</span> <span class="st">"KS"</span>, </span>
<span>                             factor_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span>, <span class="st">"mer8"</span>, <span class="st">"mer7m8"</span><span class="op">)</span>, </span>
<span>                             null_name <span class="op">=</span> <span class="st">"Background"</span>,</span>
<span>                             target_name <span class="op">=</span> <span class="st">"mer8"</span>, </span>
<span>                             alternative <span class="op">=</span> <span class="st">"greater"</span>,</span>
<span>                             height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                             width <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="SeedMatchR_files/figure-html/unnamed-chunk-18-1.png" width="480"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ensembldb_2.32.0        AnnotationFilter_1.32.0 rtracklayer_1.68.0     </span></span>
<span><span class="co">#&gt;  [4] GenomicFeatures_1.60.0  AnnotationDbi_1.70.0    Biobase_2.68.0         </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.60.0    GenomeInfoDb_1.44.0     IRanges_2.42.0         </span></span>
<span><span class="co">#&gt; [10] S4Vectors_0.46.0        BiocGenerics_0.54.0     generics_0.1.4         </span></span>
<span><span class="co">#&gt; [13] SeedMatchR_2.0.0       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1            farver_2.1.2               </span></span>
<span><span class="co">#&gt;  [3] dplyr_1.1.4                 blob_1.2.4                 </span></span>
<span><span class="co">#&gt;  [5] filelock_1.0.3              Biostrings_2.76.0          </span></span>
<span><span class="co">#&gt;  [7] bitops_1.0-9                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [9] RCurl_1.98-1.17             lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [11] BiocFileCache_2.16.0        GenomicAlignments_1.44.0   </span></span>
<span><span class="co">#&gt; [13] XML_3.99-0.18               digest_0.6.37              </span></span>
<span><span class="co">#&gt; [15] mime_0.13                   lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [17] ProtGenerics_1.40.0         KEGGREST_1.48.1            </span></span>
<span><span class="co">#&gt; [19] RSQLite_2.4.1               magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [21] compiler_4.5.1              rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [23] sass_0.4.10                 tools_4.5.1                </span></span>
<span><span class="co">#&gt; [25] gt_1.0.0                    yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [27] knitr_1.50                  labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [29] S4Arrays_1.8.1              htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [31] bit_4.6.0                   curl_6.4.0                 </span></span>
<span><span class="co">#&gt; [33] DelayedArray_0.34.1         RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [35] xml2_1.3.8                  abind_1.4-8                </span></span>
<span><span class="co">#&gt; [37] BiocParallel_1.42.1         withr_3.0.2                </span></span>
<span><span class="co">#&gt; [39] purrr_1.1.0                 desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [41] grid_4.5.1                  ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [43] scales_1.4.0                SummarizedExperiment_1.38.1</span></span>
<span><span class="co">#&gt; [45] cli_3.6.5                   rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [47] crayon_1.5.3                ragg_1.4.0                 </span></span>
<span><span class="co">#&gt; [49] httr_1.4.7                  rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [51] DBI_1.2.3                   cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [53] parallel_4.5.1              BiocManager_1.30.26        </span></span>
<span><span class="co">#&gt; [55] XVector_0.48.0              restfulr_0.0.16            </span></span>
<span><span class="co">#&gt; [57] matrixStats_1.5.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [59] Matrix_1.7-3                jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [61] bit64_4.6.0-1               systemfonts_1.2.3          </span></span>
<span><span class="co">#&gt; [63] jquerylib_0.1.4             glue_1.8.0                 </span></span>
<span><span class="co">#&gt; [65] pkgdown_2.1.3               codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [67] gtable_0.3.6                BiocVersion_3.21.1         </span></span>
<span><span class="co">#&gt; [69] BiocIO_1.18.0               UCSC.utils_1.4.0           </span></span>
<span><span class="co">#&gt; [71] tibble_3.3.0                pillar_1.11.0              </span></span>
<span><span class="co">#&gt; [73] rappdirs_0.3.3              htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [75] GenomeInfoDbData_1.2.14     R6_2.6.1                   </span></span>
<span><span class="co">#&gt; [77] dbplyr_2.5.0                textshaping_1.0.1          </span></span>
<span><span class="co">#&gt; [79] evaluate_1.0.4              lattice_0.22-7             </span></span>
<span><span class="co">#&gt; [81] AnnotationHub_3.16.0        png_0.1-8                  </span></span>
<span><span class="co">#&gt; [83] Rsamtools_2.24.0            memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [85] bslib_0.9.0                 SparseArray_1.8.0          </span></span>
<span><span class="co">#&gt; [87] xfun_0.52                   fs_1.6.6                   </span></span>
<span><span class="co">#&gt; [89] MatrixGenerics_1.20.0       pkgconfig_2.0.3</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
