<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Transcript library k-mer counts with Biostrings â€¢ SeedMatchR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Transcript library k-mer counts with Biostrings">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Transcript library k-mer counts with Biostrings</h1>
                        <h4 data-toc-skip class="author">Tareian
Cazares</h4>
            
      

      <div class="d-none name"><code>SeedMatchR_sequence_kmer_counts.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SeedMatchR</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="determine-transcript-library-k-mer-counts-with-biostrings">Determine transcript library k-mer counts with Biostrings<a class="anchor" aria-label="anchor" href="#determine-transcript-library-k-mer-counts-with-biostrings"></a>
</h2>
<p>This article will show you how to determine the transcriptome k-mer
counts for your transcriptome of interest.</p>
<div class="section level3">
<h3 id="find-hg38-7-mer-counts-across-canonical-ensembl-transcripts">Find hg38 7-mer counts across canonical ENSEMBL transcripts<a class="anchor" aria-label="anchor" href="#find-hg38-7-mer-counts-across-canonical-ensembl-transcripts"></a>
</h3>
<div class="section level4">
<h4 id="load-annotations">Load annotations<a class="anchor" aria-label="anchor" href="#load-annotations"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annodb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_annotations.html">load_annotations</a></span><span class="op">(</span><span class="st">"hg38"</span>, feature.type <span class="op">=</span> <span class="st">"exons"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Build AnnotationFilter for transcript features based on the following parameters: </span></span>
<span><span class="co">#&gt; Keep only standard chroms: TRUE</span></span>
<span><span class="co">#&gt; Remove rows with NA in transcript ID: TRUE</span></span>
<span><span class="co">#&gt; Keep only protein coding genes and transcripts: TRUE</span></span>
<span><span class="co">#&gt; Filtering for transcripts with support level: FALSE</span></span>
<span><span class="co">#&gt; Keep only the ENSEMBL canonical transcript: TRUE</span></span>
<span><span class="co">#&gt; Filtering for specific genes: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific transcripts: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific gene symbols: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific entrez id: FALSE</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span>
<span><span class="co">#&gt; Loading annotations from AnnotationHub for hg38</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("rtracklayer")</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("ensembldb")</span></span>
<span><span class="co">#&gt; Extracting exons from ensembldb object.</span></span>
<span><span class="co">#&gt; Extracting sequences for each feature.</span></span>
<span><span class="co">#&gt; Keeping sequences that are &gt;= 8</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="create-data-frame-for-mapping-tx_ix-to-gene_id">Create data frame for mapping tx_ix to gene_id<a class="anchor" aria-label="anchor" href="#create-data-frame-for-mapping-tx_ix-to-gene_id"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a data frame to map tx_id to gene_id</span></span>
<span><span class="va">tx2gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="fu">transcripts</span><span class="op">(</span><span class="va">annodb</span><span class="op">$</span><span class="va">txdb</span>, columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tx_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence LRG_432.</span></span>
<span><span class="co">#&gt;   Note that ranges located on a sequence whose length is unknown (NA) or on a</span></span>
<span><span class="co">#&gt;   circular sequence are not considered out-of-bound (use seqlengths() and</span></span>
<span><span class="co">#&gt;   isCircular() to get the lengths and circularity flags of the underlying</span></span>
<span><span class="co">#&gt;   sequences). You can use trim() to trim these ranges. See</span></span>
<span><span class="co">#&gt;   ?`trim,GenomicRanges-method` for more information.</span></span>
<span><span class="va">tx2gene</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">tx2gene</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="get-counts-data">Get counts data<a class="anchor" aria-label="anchor" href="#get-counts-data"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># only focus on ten sequences since the run can take a while during testing. </span></span>
<span><span class="va">exon.counts</span> <span class="op">=</span> <span class="fu"><a href="../reference/sequence_kmer_counts.html">sequence_kmer_counts</a></span><span class="op">(</span><span class="va">annodb</span><span class="op">$</span><span class="va">seqs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">tx2gene</span>, width <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span>
<span></span>
<span><span class="va">exon.counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;             tx_id gene_id AAAAAAA AAAAAAC AAAAAAG</span></span>
<span><span class="co">#&gt; 1 ENSG00000004059    &lt;NA&gt;       0       0       0</span></span>
<span><span class="co">#&gt; 2 ENSG00000003056    &lt;NA&gt;       6       3       0</span></span>
<span><span class="co">#&gt; 3 ENSG00000173153    &lt;NA&gt;       1       0       0</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="convert-to-k-mer-summary-table">Convert to K-mer summary table<a class="anchor" aria-label="anchor" href="#convert-to-k-mer-summary-table"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exon.binary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exon.counts</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">exon.counts</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">exon.binary</span><span class="op">[</span><span class="va">exon.binary</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">exon.gene.counts</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">exon.binary</span><span class="op">)</span></span>
<span></span>
<span><span class="va">exons.df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"exon.counts"</span> <span class="op">=</span> <span class="va">exon.gene.counts</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exons.df</span><span class="op">)</span></span>
<span><span class="co">#&gt;         exon.counts</span></span>
<span><span class="co">#&gt; AAAAAAA           5</span></span>
<span><span class="co">#&gt; AAAAAAC           3</span></span>
<span><span class="co">#&gt; AAAAAAG           4</span></span>
<span><span class="co">#&gt; AAAAAAT           4</span></span>
<span><span class="co">#&gt; AAAAACA           3</span></span>
<span><span class="co">#&gt; AAAAACC           3</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">exon.gene.counts</span>, breaks <span class="op">=</span> <span class="fl">50</span>, xlab <span class="op">=</span> <span class="st">"Number of genes with 7mer"</span>, col <span class="op">=</span> <span class="st">"grey90"</span>, main <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="SeedMatchR_sequence_kmer_counts_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
