<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Overview of SeedMatchR Annotation Databases • SeedMatchR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Overview of SeedMatchR Annotation Databases">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/SeedMatchR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/SeedMatchR_Annotation_Databases.html">Overview of SeedMatchR Annotation Databases</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_build_PPI_network.html">SeedmatchR build PPI networks</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_ECDF_statistics.html">SeedMatchR ECDF Statistics</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_noncanonical_searches.html">SeedMatchR_noncanonical_searches</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_OutputFormats.html">SeedMatchR output formats</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_search_details.html">SeedMatchR Search Details</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchR_sequence_kmer_counts.html">Transcript library k-mer counts with Biostrings</a></li>
    <li><a class="dropdown-item" href="../articles/SeedMatchReport.html">Generate a SeedMatchReport for select sequence definitions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Overview of SeedMatchR Annotation Databases</h1>
                        <h4 data-toc-skip class="author">Tareian
Cazares</h4>
            
      

      <div class="d-none name"><code>SeedMatchR_Annotation_Databases.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This entry covers which annotations are available for the 3 model
species used in <code>SeedMatchR</code> (human, mouse, rat).
<code>SeedMatchR</code> requires information on the GTF for gene &gt;
transcript annotations as well as the sequences that correspond to the
features of interest being searched. These can be generated
independently by the user or they can be loaded with the
<code><a href="../reference/load_annotations.html">SeedMatchR::load_annotations</a></code> function. This function will
load the predefined annotations based on the species of interest, the
filtering criteria, and features of interest.</p>
<p>NOTES:</p>
<ol style="list-style-type: decimal">
<li>In order to use <code>rnor7</code>, you must have R 4.3.0 or greater
installed. This is because <code>AnnotationHub</code> ≥ 3.8.0 is
required.</li>
<li>
<code>rnor6</code> and <code>mm10</code> cannot be used with the
<code><a href="https://rdrr.io/pkg/ensembldb/man/Filter-classes.html" class="external-link">ensembldb::TxIsCanonicalFilter</a></code> therefore the
<code>canonical</code> argument for
<code><a href="../reference/load_annotations.html">SeedMatchR::load_annotations</a></code> must be set to
<code>FALSE</code>.</li>
<li>
<code>rnor6</code> and <code>rnor7</code> cannot be used with
<code><a href="https://rdrr.io/pkg/ensembldb/man/Filter-classes.html" class="external-link">ensembldb::TxSupportLevelFilter</a></code>. Therefore this value
should not be passed to the function
<code><a href="../reference/load_annotations.html">SeedMatchR::load_annotations</a></code> when working with the rat
transcriptome.</li>
</ol>
</div>
<div class="section level2">
<h2 id="current-annotation-options">Current Annotation Options<a class="anchor" aria-label="anchor" href="#current-annotation-options"></a>
</h2>
<table class="table">
<colgroup>
<col width="9%">
<col width="17%">
<col width="18%">
<col width="16%">
<col width="16%">
<col width="10%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th>Species</th>
<th>SeedMatchR arg</th>
<th>Reference Build</th>
<th>ENSEMBL Build</th>
<th>Release Date</th>
<th>2bit ID</th>
<th>ensembldb</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Human</td>
<td>hg38</td>
<td>GRCh38.p13</td>
<td>109</td>
<td>February 2023</td>
<td>AH106283</td>
<td>AH109606</td>
</tr>
<tr class="even">
<td>Rat</td>
<td>rnor6</td>
<td>Rnor_v6.0</td>
<td>104</td>
<td>May 2011</td>
<td>AH93578</td>
<td>AH95846</td>
</tr>
<tr class="odd">
<td>Rat</td>
<td>rnor7</td>
<td>Rnor_v7.2</td>
<td>109</td>
<td>February 2023</td>
<td>AH106786</td>
<td>AH109732</td>
</tr>
<tr class="even">
<td>Rat</td>
<td>rnor7.113</td>
<td>Rnor_v7.2</td>
<td>113</td>
<td>October 2024</td>
<td>AH106786</td>
<td>AH119437</td>
</tr>
<tr class="odd">
<td>Mouse</td>
<td>mm39</td>
<td>GRCm39</td>
<td>109</td>
<td>February 2023</td>
<td>AH106440</td>
<td>AH109655</td>
</tr>
<tr class="even">
<td>Mouse</td>
<td>mm10</td>
<td>GRCm38</td>
<td>102</td>
<td>October 2020</td>
<td>AH88475</td>
<td>AH89211</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="functions-for-working-with-annotations-and-building-transcriptomes-for-queries">Functions for working with annotations and building transcriptomes
for queries<a class="anchor" aria-label="anchor" href="#functions-for-working-with-annotations-and-building-transcriptomes-for-queries"></a>
</h2>
<div class="section level3">
<h3 id="load-data-for-rnor7">Load data for Rnor7<a class="anchor" aria-label="anchor" href="#load-data-for-rnor7"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annodb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_annotations.html">load_annotations</a></span><span class="op">(</span><span class="st">"rnor7"</span>, feature.type <span class="op">=</span> <span class="st">"exons"</span>, protein.coding <span class="op">=</span> <span class="cn">FALSE</span>, canonical <span class="op">=</span> <span class="cn">F</span>, return_gene_name <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="co">#&gt; Build AnnotationFilter for transcript features based on the following parameters: </span></span>
<span><span class="co">#&gt; Keep only standard chroms: TRUE</span></span>
<span><span class="co">#&gt; Remove rows with NA in transcript ID: TRUE</span></span>
<span><span class="co">#&gt; Keep only protein coding genes and transcripts: FALSE</span></span>
<span><span class="co">#&gt; Filtering for transcripts with support level: FALSE</span></span>
<span><span class="co">#&gt; Keep only the ENSEMBL canonical transcript: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific genes: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific transcripts: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific gene symbols: FALSE</span></span>
<span><span class="co">#&gt; Filtering for specific entrez id: FALSE</span></span>
<span><span class="co">#&gt; Loading annotations from AnnotationHub for rnor7</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("rtracklayer")</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span>
<span><span class="co">#&gt; downloading 1 resources</span></span>
<span><span class="co">#&gt; retrieving 1 resource</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("ensembldb")</span></span>
<span><span class="co">#&gt; Extracting exons from ensembldb object.</span></span>
<span><span class="co">#&gt; Extracting sequences for each feature.</span></span>
<span><span class="co">#&gt; Keeping sequences that are &gt;= 8</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="how-annotations-were-retrieved">How annotations were retrieved<a class="anchor" aria-label="anchor" href="#how-annotations-were-retrieved"></a>
</h2>
<div class="section level3">
<h3 id="selecting-gtf">Selecting GTF<a class="anchor" aria-label="anchor" href="#selecting-gtf"></a>
</h3>
<p>The GTF files were selected to have the full DNA sequence visible
without any type of masking of repeats. These files have names like
<code>Mus_musculus.GRCm39.109.gtf</code>.</p>
</div>
<div class="section level3">
<h3 id="selecting--2bit-dna-sequences">Selecting <code>.2bit</code> DNA sequences<a class="anchor" aria-label="anchor" href="#selecting--2bit-dna-sequences"></a>
</h3>
<p>The <code>.2bit</code> files were selected to have the full DNA
sequence visible without any type of masking of repeats. These files
have names like
<code>Homo_sapiens.GRCh38.dna.primary_assembly.2bit</code>. For Rat,
there was no file that was named as the
<code>dna.primary_assembly</code>, instead we had to choose from
<code>dna_sm.primary_assembly</code> which is the soft-masked version of
the genome that accounts for repeats.</p>
</div>
</div>
<div class="section level2">
<h2 id="code-for-querying-annotationhub-for-references">Code for querying <code>AnnotationHub</code> for references<a class="anchor" aria-label="anchor" href="#code-for-querying-annotationhub-for-references"></a>
</h2>
<div class="section level3">
<h3 id="hg38">Hg38<a class="anchor" aria-label="anchor" href="#hg38"></a>
</h3>
<div class="section level4">
<h4 id="bit">
<code>.2bit</code><a class="anchor" aria-label="anchor" href="#bit"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCh38"</span>, <span class="st">"2bit"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 120 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH49722"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                           </span></span>
<span><span class="co">#&gt;   AH49722  | Homo_sapiens.GRCh38.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH49723  | Homo_sapiens.GRCh38.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH49724  | Homo_sapiens.GRCh38.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH49725  | Homo_sapiens.GRCh38.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH49726  | Homo_sapiens.GRCh38.ncrna.2bit                  </span></span>
<span><span class="co">#&gt;   ...        ...                                             </span></span>
<span><span class="co">#&gt;   AH106282 | Homo_sapiens.GRCh38.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH106283 | Homo_sapiens.GRCh38.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH106284 | Homo_sapiens.GRCh38.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH106285 | Homo_sapiens.GRCh38.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH106286 | Homo_sapiens.GRCh38.ncrna.2bit</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="ensembldb">
<code>ensembldb</code><a class="anchor" aria-label="anchor" href="#ensembldb"></a>
</h4>
<div class="section level5">
<h5 id="version-110-for-grch38-p13">version 110 for GRCh38.p13<a class="anchor" aria-label="anchor" href="#version-110-for-grch38-p13"></a>
</h5>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCh38"</span>, <span class="st">"110"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # names(): AH113665</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2023-04-25</span></span>
<span><span class="co">#&gt; # $title: Ensembl 110 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt; # $description: Gene and protein annotations for Homo sapiens based on Ensem...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh38</span></span>
<span><span class="co">#&gt; # $sourcetype: ensembl</span></span>
<span><span class="co">#&gt; # $sourceurl: http://www.ensembl.org</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("110", "Annotation", "AnnotationHubSoftware", "Coverage",</span></span>
<span><span class="co">#&gt; #   "DataImport", "EnsDb", "Ensembl", "Gene", "Protein", "Sequencing",</span></span>
<span><span class="co">#&gt; #   "Transcript") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH113665"]]'</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="version-80-for-grch38-p2">Version 80 for GRCh38.p2<a class="anchor" aria-label="anchor" href="#version-80-for-grch38-p2"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCh38"</span>, <span class="st">"80"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # names(): AH47066</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2015-05-22</span></span>
<span><span class="co">#&gt; # $title: Homo_sapiens.GRCh38.80.gtf</span></span>
<span><span class="co">#&gt; # $description: Gene Annotation for Homo sapiens</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh38</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: ftp://ftp.ensembl.org/pub/release-80/gtf/homo_sapiens/Homo_sap...</span></span>
<span><span class="co">#&gt; # $sourcesize: 44733199</span></span>
<span><span class="co">#&gt; # $tags: c("GTF", "ensembl", "Gene", "Transcript", "Annotation") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH47066"]]'</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="rat">Rat<a class="anchor" aria-label="anchor" href="#rat"></a>
</h3>
<div class="section level4">
<h4 id="bit-1">
<code>.2bit</code><a class="anchor" aria-label="anchor" href="#bit-1"></a>
</h4>
<div class="section level5">
<h5 id="rnor6">Rnor6<a class="anchor" aria-label="anchor" href="#rnor6"></a>
</h5>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Rnor"</span>, <span class="st">"release-104"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 7 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: rattus norvegicus</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile, GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH92418"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             title                                          </span></span>
<span><span class="co">#&gt;   AH92418 | Rattus_norvegicus.Rnor_6.0.104.abinitio.gtf    </span></span>
<span><span class="co">#&gt;   AH92419 | Rattus_norvegicus.Rnor_6.0.104.chr.gtf         </span></span>
<span><span class="co">#&gt;   AH92420 | Rattus_norvegicus.Rnor_6.0.104.gtf             </span></span>
<span><span class="co">#&gt;   AH93576 | Rattus_norvegicus.Rnor_6.0.cdna.all.2bit       </span></span>
<span><span class="co">#&gt;   AH93577 | Rattus_norvegicus.Rnor_6.0.dna_rm.toplevel.2bit</span></span>
<span><span class="co">#&gt;   AH93578 | Rattus_norvegicus.Rnor_6.0.dna_sm.toplevel.2bit</span></span>
<span><span class="co">#&gt;   AH93579 | Rattus_norvegicus.Rnor_6.0.ncrna.2bit</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="rnor7">Rnor7<a class="anchor" aria-label="anchor" href="#rnor7"></a>
</h5>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Rattus"</span>, <span class="st">".2bit"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 106 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl, UCSC</span></span>
<span><span class="co">#&gt; # $species: Rattus norvegicus, rattus norvegicus</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH14021"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                           </span></span>
<span><span class="co">#&gt;   AH14021  | rn6.2bit                                        </span></span>
<span><span class="co">#&gt;   AH14022  | rn5.2bit                                        </span></span>
<span><span class="co">#&gt;   AH14023  | rn4.2bit                                        </span></span>
<span><span class="co">#&gt;   AH49867  | Rattus_norvegicus.Rnor_6.0.cdna.all.2bit        </span></span>
<span><span class="co">#&gt;   AH49868  | Rattus_norvegicus.Rnor_6.0.dna_rm.toplevel.2bit </span></span>
<span><span class="co">#&gt;   ...        ...                                             </span></span>
<span><span class="co">#&gt;   AH104348 | Rattus_norvegicus.mRatBN7.2.ncrna.2bit          </span></span>
<span><span class="co">#&gt;   AH106784 | Rattus_norvegicus.mRatBN7.2.cdna.all.2bit       </span></span>
<span><span class="co">#&gt;   AH106785 | Rattus_norvegicus.mRatBN7.2.dna_rm.toplevel.2bit</span></span>
<span><span class="co">#&gt;   AH106786 | Rattus_norvegicus.mRatBN7.2.dna_sm.toplevel.2bit</span></span>
<span><span class="co">#&gt;   AH106787 | Rattus_norvegicus.mRatBN7.2.ncrna.2bit</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="ensembldb-1">
<code>ensembldb</code><a class="anchor" aria-label="anchor" href="#ensembldb-1"></a>
</h4>
<div class="section level5">
<h5 id="rnor6-1">Rnor6<a class="anchor" aria-label="anchor" href="#rnor6-1"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rat"</span>, <span class="fl">104</span>, <span class="st">"ensdb"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 6 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Sparus aurata, Rattus norvegicus, Mesocricetus auratus, Kryptole...</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH95677"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             title                                        </span></span>
<span><span class="co">#&gt;   AH95677 | Ensembl 104 EnsDb for Carassius auratus      </span></span>
<span><span class="co">#&gt;   AH95725 | Ensembl 104 EnsDb for Echeneis naucrates     </span></span>
<span><span class="co">#&gt;   AH95749 | Ensembl 104 EnsDb for Kryptolebias marmoratus</span></span>
<span><span class="co">#&gt;   AH95762 | Ensembl 104 EnsDb for Mesocricetus auratus   </span></span>
<span><span class="co">#&gt;   AH95846 | Ensembl 104 EnsDb for Rattus norvegicus      </span></span>
<span><span class="co">#&gt;   AH95850 | Ensembl 104 EnsDb for Sparus aurata</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="rnor7-1">Rnor7<a class="anchor" aria-label="anchor" href="#rnor7-1"></a>
</h5>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rat"</span>, <span class="fl">109</span>, <span class="st">"ensdb"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 6 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Sparus aurata, Rattus norvegicus, Mesocricetus auratus, Kryptole...</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH109518"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                        </span></span>
<span><span class="co">#&gt;   AH109518 | Ensembl 109 EnsDb for Carassius auratus      </span></span>
<span><span class="co">#&gt;   AH109583 | Ensembl 109 EnsDb for Echeneis naucrates     </span></span>
<span><span class="co">#&gt;   AH109611 | Ensembl 109 EnsDb for Kryptolebias marmoratus</span></span>
<span><span class="co">#&gt;   AH109625 | Ensembl 109 EnsDb for Mesocricetus auratus   </span></span>
<span><span class="co">#&gt;   AH109732 | Ensembl 109 EnsDb for Rattus norvegicus      </span></span>
<span><span class="co">#&gt;   AH109736 | Ensembl 109 EnsDb for Sparus aurata</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="mouse">Mouse<a class="anchor" aria-label="anchor" href="#mouse"></a>
</h3>
<div class="section level4">
<h4 id="bit-2">
<code>.2bit</code><a class="anchor" aria-label="anchor" href="#bit-2"></a>
</h4>
<div class="section level5">
<h5 id="mm39-or-gcrm39">mm39 or GCRm39<a class="anchor" aria-label="anchor" href="#mm39-or-gcrm39"></a>
</h5>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCm39"</span>, <span class="st">".2bit"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 25 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH90962"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                           </span></span>
<span><span class="co">#&gt;   AH90962  | Mus_musculus.GRCm39.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH90963  | Mus_musculus.GRCm39.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH90964  | Mus_musculus.GRCm39.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH90965  | Mus_musculus.GRCm39.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH90966  | Mus_musculus.GRCm39.ncrna.2bit                  </span></span>
<span><span class="co">#&gt;   ...        ...                                             </span></span>
<span><span class="co">#&gt;   AH106439 | Mus_musculus.GRCm39.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH106440 | Mus_musculus.GRCm39.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH106441 | Mus_musculus.GRCm39.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH106442 | Mus_musculus.GRCm39.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH106443 | Mus_musculus.GRCm39.ncrna.2bit</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="mm10-or-grcm38">mm10 or GRCm38<a class="anchor" aria-label="anchor" href="#mm10-or-grcm38"></a>
</h5>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCm38"</span>, <span class="st">".2bit"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 95 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Mus musculus, mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: TwoBitFile</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH49772"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             title                                           </span></span>
<span><span class="co">#&gt;   AH49772 | Mus_musculus.GRCm38.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH49773 | Mus_musculus.GRCm38.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH49774 | Mus_musculus.GRCm38.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH49775 | Mus_musculus.GRCm38.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH49776 | Mus_musculus.GRCm38.ncrna.2bit                  </span></span>
<span><span class="co">#&gt;   ...       ...                                             </span></span>
<span><span class="co">#&gt;   AH88474 | Mus_musculus.GRCm38.cdna.all.2bit               </span></span>
<span><span class="co">#&gt;   AH88475 | Mus_musculus.GRCm38.dna.primary_assembly.2bit   </span></span>
<span><span class="co">#&gt;   AH88476 | Mus_musculus.GRCm38.dna_rm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH88477 | Mus_musculus.GRCm38.dna_sm.primary_assembly.2bit</span></span>
<span><span class="co">#&gt;   AH88478 | Mus_musculus.GRCm38.ncrna.2bit</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="ensembldb-2">
<code>ensembldb</code><a class="anchor" aria-label="anchor" href="#ensembldb-2"></a>
</h4>
<div class="section level5">
<h5 id="mm39-or-grcm39">mm39 or GRCm39<a class="anchor" aria-label="anchor" href="#mm39-or-grcm39"></a>
</h5>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mus"</span>, <span class="fl">104</span>, <span class="st">"ensdb"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 11 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Ursus maritimus, Scophthalmus maximus, Prolemur simus, Periophth...</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH95670"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             title                                              </span></span>
<span><span class="co">#&gt;   AH95670 | Ensembl 104 EnsDb for Balaenoptera musculus        </span></span>
<span><span class="co">#&gt;   AH95763 | Ensembl 104 EnsDb for Mus caroli                   </span></span>
<span><span class="co">#&gt;   AH95775 | Ensembl 104 EnsDb for Mus musculus                 </span></span>
<span><span class="co">#&gt;   AH95778 | Ensembl 104 EnsDb for Mus pahari                   </span></span>
<span><span class="co">#&gt;   AH95779 | Ensembl 104 EnsDb for Mus spicilegus               </span></span>
<span><span class="co">#&gt;   ...       ...                                                </span></span>
<span><span class="co">#&gt;   AH95790 | Ensembl 104 EnsDb for Neogobius melanostomus       </span></span>
<span><span class="co">#&gt;   AH95825 | Ensembl 104 EnsDb for Periophthalmus magnuspinnatus</span></span>
<span><span class="co">#&gt;   AH95836 | Ensembl 104 EnsDb for Prolemur simus               </span></span>
<span><span class="co">#&gt;   AH95861 | Ensembl 104 EnsDb for Scophthalmus maximus         </span></span>
<span><span class="co">#&gt;   AH95879 | Ensembl 104 EnsDb for Ursus maritimus</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="mm10-or-grcm38-1">mm10 or GRCm38<a class="anchor" aria-label="anchor" href="#mm10-or-grcm38-1"></a>
</h5>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">query</span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mus"</span>, <span class="fl">109</span>, <span class="st">"ensdb"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 27 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2025-04-08</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Mus musculus, Ursus maritimus, Scophthalmus maximus, Prolemur si...</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH109509"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                              </span></span>
<span><span class="co">#&gt;   AH109509 | Ensembl 109 EnsDb for Balaenoptera musculus        </span></span>
<span><span class="co">#&gt;   AH109626 | Ensembl 109 EnsDb for Mus caroli                   </span></span>
<span><span class="co">#&gt;   AH109640 | Ensembl 109 EnsDb for Mus musculus                 </span></span>
<span><span class="co">#&gt;   AH109641 | Ensembl 109 EnsDb for Mus musculus                 </span></span>
<span><span class="co">#&gt;   AH109642 | Ensembl 109 EnsDb for Mus musculus                 </span></span>
<span><span class="co">#&gt;   ...        ...                                                </span></span>
<span><span class="co">#&gt;   AH109671 | Ensembl 109 EnsDb for Neogobius melanostomus       </span></span>
<span><span class="co">#&gt;   AH109709 | Ensembl 109 EnsDb for Periophthalmus magnuspinnatus</span></span>
<span><span class="co">#&gt;   AH109721 | Ensembl 109 EnsDb for Prolemur simus               </span></span>
<span><span class="co">#&gt;   AH109750 | Ensembl 109 EnsDb for Scophthalmus maximus         </span></span>
<span><span class="co">#&gt;   AH109783 | Ensembl 109 EnsDb for Ursus maritimus</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ensembldb_2.32.0        AnnotationFilter_1.32.0 GenomicFeatures_1.60.0 </span></span>
<span><span class="co">#&gt;  [4] AnnotationDbi_1.70.0    Biobase_2.68.0          rtracklayer_1.68.0     </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.60.0    GenomeInfoDb_1.44.0     IRanges_2.42.0         </span></span>
<span><span class="co">#&gt; [10] S4Vectors_0.46.0        SeedMatchR_2.0.0        AnnotationHub_3.16.0   </span></span>
<span><span class="co">#&gt; [13] BiocFileCache_2.16.0    dbplyr_2.5.0            BiocGenerics_0.54.0    </span></span>
<span><span class="co">#&gt; [16] generics_0.1.4         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1            dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [3] blob_1.2.4                  filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [5] Biostrings_2.76.0           bitops_1.0-9               </span></span>
<span><span class="co">#&gt;  [7] fastmap_1.2.0               lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [9] RCurl_1.98-1.17             GenomicAlignments_1.44.0   </span></span>
<span><span class="co">#&gt; [11] XML_3.99-0.18               digest_0.6.37              </span></span>
<span><span class="co">#&gt; [13] mime_0.13                   lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [15] ProtGenerics_1.40.0         KEGGREST_1.48.1            </span></span>
<span><span class="co">#&gt; [17] RSQLite_2.4.1               magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [19] compiler_4.5.1              rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [21] sass_0.4.10                 tools_4.5.1                </span></span>
<span><span class="co">#&gt; [23] yaml_2.3.10                 knitr_1.50                 </span></span>
<span><span class="co">#&gt; [25] S4Arrays_1.8.1              htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [27] bit_4.6.0                   curl_6.4.0                 </span></span>
<span><span class="co">#&gt; [29] DelayedArray_0.34.1         abind_1.4-8                </span></span>
<span><span class="co">#&gt; [31] BiocParallel_1.42.1         withr_3.0.2                </span></span>
<span><span class="co">#&gt; [33] purrr_1.1.0                 desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [35] grid_4.5.1                  SummarizedExperiment_1.38.1</span></span>
<span><span class="co">#&gt; [37] cli_3.6.5                   rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [39] crayon_1.5.3                ragg_1.4.0                 </span></span>
<span><span class="co">#&gt; [41] httr_1.4.7                  rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [43] DBI_1.2.3                   cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [45] parallel_4.5.1              BiocManager_1.30.26        </span></span>
<span><span class="co">#&gt; [47] XVector_0.48.0              restfulr_0.0.16            </span></span>
<span><span class="co">#&gt; [49] matrixStats_1.5.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [51] Matrix_1.7-3                jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [53] bit64_4.6.0-1               systemfonts_1.2.3          </span></span>
<span><span class="co">#&gt; [55] jquerylib_0.1.4             glue_1.8.0                 </span></span>
<span><span class="co">#&gt; [57] pkgdown_2.1.3               codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [59] BiocVersion_3.21.1          BiocIO_1.18.0              </span></span>
<span><span class="co">#&gt; [61] UCSC.utils_1.4.0            tibble_3.3.0               </span></span>
<span><span class="co">#&gt; [63] pillar_1.11.0               rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt; [65] htmltools_0.5.8.1           GenomeInfoDbData_1.2.14    </span></span>
<span><span class="co">#&gt; [67] R6_2.6.1                    textshaping_1.0.1          </span></span>
<span><span class="co">#&gt; [69] lattice_0.22-7              evaluate_1.0.4             </span></span>
<span><span class="co">#&gt; [71] png_0.1-8                   Rsamtools_2.24.0           </span></span>
<span><span class="co">#&gt; [73] memoise_2.0.1               bslib_0.9.0                </span></span>
<span><span class="co">#&gt; [75] SparseArray_1.8.0           xfun_0.52                  </span></span>
<span><span class="co">#&gt; [77] fs_1.6.6                    MatrixGenerics_1.20.0      </span></span>
<span><span class="co">#&gt; [79] pkgconfig_2.0.3</span></span></code></pre></div>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tareian Cazares, Gulcin Ozer, Pradeep Varathan Pugalenthi, Jibo Wang, Rick Higgs.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
