<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SeedMatchR">
<title>Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR • SeedMatchR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR">
<meta property="og:description" content="SeedMatchR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SeedMatchR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/SeedMatchR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/SeedMatchR_miRNA_analysis.html">Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Exploring perturbation of miRNA pathways by siRNA knockdown using SeedMatchR</h1>
            
      
      
      <div class="d-none name"><code>SeedMatchR_miRNA_analysis.Rmd</code></div>
    </div>

    
    
<p>In this example we will use a few <code>SeedMatchR</code> functions
to examine changes in expression profiles of miRNA target genes. We will
use miRDB as our source of miRNA targets.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of SeedMatchR from <a href="https://github.com/" class="external-link">GitHub</a> or the stable build from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tacazares/SeedMatchR"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://turbo-adventure-19v5gn5.pages.github.io/" class="external-link">SeedMatchR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h2>
<p>This example uses the siRNA sequence, D1, targeting the Ttr gene in
rat liver from the publication:</p>
<pre><code>Schlegel MK, Janas MM, Jiang Y, Barry JD, Davis W, Agarwal S, Berman D, Brown CR, Castoreno A, LeBlanc S, Liebow A, Mayo T, Milstein S, Nguyen T, Shulga-Morskaya S, Hyde S, Schofield S, Szeto J, Woods LB, Yilmaz VO, Manoharan M, Egli M, Charissé K, Sepp-Lorenzino L, Haslett P, Fitzgerald K, Jadhav V, Maier MA. From bench to bedside: Improving the clinical safety of GalNAc-siRNA conjugates using seed-pairing destabilization. Nucleic Acids Res. 2022 Jul 8;50(12):6656-6670. doi: 10.1093/nar/gkac539. PMID: 35736224; PMCID: PMC9262600.</code></pre>
<p>The guide sequence of interest is 23 bp long and oriented 5’ -&gt;
3’.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># siRNA sequence of interest targeting a 23 bp region of the Ttr gene</span></span>
<span><span class="va">guide.seq</span> <span class="op">=</span> <span class="st">"UUAUAGAGCAAGAACACUGUUUU"</span></span></code></pre></div>
<div class="section level4">
<h4 id="download-data-only-need-to-perform-once-can-skip-to-loading-if-done">Download data (only need to perform once, can skip to loading if
done)<a class="anchor" aria-label="anchor" href="#download-data-only-need-to-perform-once-can-skip-to-loading-if-done"></a>
</h4>
<p>We start by downloading the example data set. This function will
download three files from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE184929" class="external-link">GSE184929</a>.
These files represent three samples with different siRNA treatments at
two dosages.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"mirna"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-example-data-1">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data-1"></a>
</h4>
<p>We can load the example data into the environment.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"mirna"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"sirna"</span><span class="op">)</span></span></code></pre></div>
<p>The DESeq2 results are available through the names
<code>Schlegel_2022_Ttr_D1_30mkg</code>,
<code>Schlegel_2022_Ttr_D4_30mkg</code> and
<code>Schlegel_2022_Ttr_D1_10mkg</code>. The data set name is long, so
it will be renamed to <code>res</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">Schlegel_2022_Ttr_D1_10mkg</span></span>
<span></span>
<span><span class="co"># Filter DESeq2 results for SeedMatchR</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_deseq.html">filter_deseq</a></span><span class="op">(</span><span class="va">res</span>, fdr.cutoff<span class="op">=</span><span class="fl">1</span>, fc.cutoff<span class="op">=</span><span class="fl">0</span>, rm.na.log2fc <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="load-annotations">Load Annotations<a class="anchor" aria-label="anchor" href="#load-annotations"></a>
</h2>
<p>We will next load the annotations for rat. This step is further
described in the vignette.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the species specific annotation database object</span></span>
<span><span class="va">anno.db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_species_anno_db.html">load_species_anno_db</a></span><span class="op">(</span><span class="st">"rat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the specific features and sequences of interest</span></span>
<span><span class="va">features</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_feature_seqs.html">get_feature_seqs</a></span><span class="op">(</span><span class="va">anno.db</span><span class="op">$</span><span class="va">tx.db</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">dna</span>, feature.type <span class="op">=</span> <span class="st">"3UTR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="search-features-for-seed-matches">Search features for seed matches<a class="anchor" aria-label="anchor" href="#search-features-for-seed-matches"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/SeedMatchR.html">SeedMatchR</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">anno.db</span><span class="op">$</span><span class="va">gtf</span>, <span class="va">features</span><span class="op">$</span><span class="va">seqs</span>, <span class="va">guide.seq</span>, <span class="st">"mer7m8"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id   baseMean log2FoldChange     lfcSE       stat</span></span>
<span><span class="co">#&gt; 1 ENSRNOG00000016275 4091.69884     -8.4473574        NA -30.515742</span></span>
<span><span class="co">#&gt; 2 ENSRNOG00000042237   65.09709     -2.0505223 0.1928097 -10.634955</span></span>
<span><span class="co">#&gt; 3 ENSRNOG00000000127  473.07585     -0.9871468 0.1079412  -9.145227</span></span>
<span><span class="co">#&gt; 4 ENSRNOG00000047179 2033.36328     -0.7260060 0.1044405  -6.951386</span></span>
<span><span class="co">#&gt; 5 ENSRNOG00000019525 2064.65901     -0.6957186 0.1087750  -6.395941</span></span>
<span><span class="co">#&gt; 6 ENSRNOG00000004821   68.37800      1.1960943 0.1957843   6.109246</span></span>
<span><span class="co">#&gt;          pvalue          padj symbol mer7m8</span></span>
<span><span class="co">#&gt; 1 1.611228e-204 1.223406e-200    Ttr      1</span></span>
<span><span class="co">#&gt; 2  2.049278e-26  7.780083e-23 Tm6sf2      0</span></span>
<span><span class="co">#&gt; 3  5.950483e-20  1.506067e-16  Kpna6      0</span></span>
<span><span class="co">#&gt; 4  3.617155e-12  6.866265e-09  Aplp2      1</span></span>
<span><span class="co">#&gt; 5  1.595611e-10  2.423095e-07  Hspa9      0</span></span>
<span><span class="co">#&gt; 6  1.001031e-09  1.266804e-06  Sntb1      0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-mirdb-for-use-with-seedmatchr">Preparing miRDB for use with SeedMatchR<a class="anchor" aria-label="anchor" href="#preparing-mirdb-for-use-with-seedmatchr"></a>
</h2>
<p>The <code>miRDB</code> database is used to identify mRNA that are
predicted to be targets of miRNA. We download the latest version of
miRDB from the website and modify the data to include information for
gene symbol, ENTREZ ID, and ENSEMBL ID. We use the
<code>org.Rn.eg.db</code> data package to access species specific
annotation information for ENSEMBL.</p>
<div class="section level3">
<h3 id="download-mirdb-only-needs-to-be-performed-once-can-skip-to-loading-if-done">Download miRDB (only needs to be performed once, can skip to loading
if done)<a class="anchor" aria-label="anchor" href="#download-mirdb-only-needs-to-be-performed-once-can-skip-to-loading-if-done"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_example_data.html">get_example_data</a></span><span class="op">(</span><span class="st">"mirna"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in dir.create(data.path, recursive = TRUE):</span></span>
<span><span class="co">#&gt; '/home/runner/.local/share/R/SeedMatchR' already exists</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-mirdb-into-the-environment">Load miRDB into the environment<a class="anchor" aria-label="anchor" href="#load-mirdb-into-the-environment"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="st">"mirna"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mirdb.rat</span> <span class="op">=</span> <span class="va">mirdb</span><span class="op">[</span><span class="va">mirdb</span><span class="op">$</span><span class="va">species</span> <span class="op">==</span> <span class="st">"rno"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mirdb.rat</span><span class="op">)</span></span>
<span><span class="co">#&gt;                miRDB.ID target.REFSEQ.ID target.score species  miRNA.name</span></span>
<span><span class="co">#&gt; 6630977 rno-let-7a-1-3p     NM_001105825     56.73140     rno let-7a-1-3p</span></span>
<span><span class="co">#&gt; 6630978 rno-let-7a-1-3p     NM_001025710     96.90071     rno let-7a-1-3p</span></span>
<span><span class="co">#&gt; 6630979 rno-let-7a-1-3p        NM_022257     80.13760     rno let-7a-1-3p</span></span>
<span><span class="co">#&gt; 6630980 rno-let-7a-1-3p     NM_001107001     85.45625     rno let-7a-1-3p</span></span>
<span><span class="co">#&gt; 6630981 rno-let-7a-1-3p     NM_001037184     93.84484     rno let-7a-1-3p</span></span>
<span><span class="co">#&gt; 6630982 rno-let-7a-1-3p     NM_001127552     72.10032     rno let-7a-1-3p</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="map-ensembl-names">Map ENSEMBL names<a class="anchor" aria-label="anchor" href="#map-ensembl-names"></a>
</h3>
<p>Since the names of the target genes are in REFSEQ format, we will use
the R package <code>org.Rn.eg.db</code> to help map REFSEQ IDs to
ENSEMBL IDs.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Rn.eg.db</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationDbi</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:AnnotationDbi':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     select</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     combine</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:GenomicRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, union</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:GenomeInfoDb':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:IRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     collapse, desc, intersect, setdiff, slice, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     combine, intersect, setdiff, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span></span>
<span><span class="co"># Create the annotation database based on refseq, ensembl, and gene symbol annotations</span></span>
<span><span class="va">mir.targets</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Rn.eg.db</span>,</span>
<span>                                      keys <span class="op">=</span> <span class="va">mirdb.rat</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                                      keytype <span class="op">=</span><span class="st">"REFSEQ"</span>,</span>
<span>                                      columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"SYMBOL"</span>, <span class="st">"ENTREZID"</span><span class="op">)</span>,</span>
<span>                                      multiVals<span class="op">=</span><span class="st">"list"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned many:many mapping between keys and columns</span></span>
<span></span>
<span>  <span class="co"># Find the IDs as as ENSEMBL IDs and return as a list</span></span>
<span><span class="va">mir.targ.ens</span> <span class="op">&lt;-</span> <span class="va">mir.targets</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">REFSEQ</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>ENSEMBL_ID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/toString.html" class="external-link">toString</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ENSEMBL</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Match the IDs for ENSEMBL</span></span>
<span><span class="va">ref2ens</span> <span class="op">&lt;-</span> <span class="va">mir.targ.ens</span><span class="op">$</span><span class="va">ENSEMBL_ID</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">mirdb.rat</span><span class="op">$</span><span class="va">target.REFSEQ.ID</span>,</span>
<span>                                           <span class="va">mir.targ.ens</span><span class="op">$</span><span class="va">REFSEQ</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Bind the new columns</span></span>
<span><span class="va">mirdb.rat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">mirdb.rat</span>, <span class="va">ref2ens</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="finding-mirna-target-genes-based-on-mirna-expression-data">Finding miRNA target genes based on miRNA expression data<a class="anchor" aria-label="anchor" href="#finding-mirna-target-genes-based-on-mirna-expression-data"></a>
</h2>
<p>If you have smallRNAseq expression data, you can use it to decide
which miRNA are potentially expressed in your system. If not, you can
follow the below approach to identify public small RNAseq experiments
and generate a list of top expressed microRNA.</p>
<div class="section level3">
<h3 id="public-rat-liver-small-rna-expression-data">Public rat liver small RNA expression data<a class="anchor" aria-label="anchor" href="#public-rat-liver-small-rna-expression-data"></a>
</h3>
<p>Here is an example of looking at miRNA expression data from 6 week
old female rat liver in baseline conditions. This data was collected
from the GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE172269" class="external-link">GSE172269</a>.
There are four replicates GSM5251324, GSM5251325, GSM5251326,
GSM5251327.</p>
<p>In this case, we work with the normalized counts data. We take the
average across all replicates in the group to find the average miRNA
expression. Then we sort the list by highest expressed miRNA. You can
then choose a cutoff based off of expression, rank, or some combination
of the two. This data is provided as a data object called
<code>GSE172269_F_Lvr_6wks_miRNA</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_10</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rno-miR-192-5p"</span>, <span class="st">"rno-miR-22-3p"</span>, <span class="st">"rno-miR-148a-3p"</span>, <span class="st">"rno-miR-10a-5p"</span>, <span class="st">"rno-miR-26a-5p"</span>, <span class="st">"rno-miR-21-5p"</span>, <span class="st">"rno-miR-143-3p"</span>, <span class="st">"rno-miR-27b-3p"</span>, <span class="st">"rno-miR-191a-5p"</span>, <span class="st">"rno-miR-122-5p"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># subset the database according to the miRNA of interest and the target score</span></span>
<span><span class="va">mir.targets.score90</span> <span class="op">=</span> <span class="va">mirdb.rat</span><span class="op">$</span><span class="va">ref2ens</span><span class="op">[</span><span class="va">mirdb.rat</span><span class="op">[</span>,<span class="st">"miRDB.ID"</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top_10</span> <span class="op">&amp;</span> <span class="va">mirdb.rat</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">&gt;=</span> <span class="fl">90</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Remove any NA values</span></span>
<span><span class="va">mir.targets.score90</span> <span class="op">=</span> <span class="va">mir.targets.score90</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">mir.targets.score90</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># unlist the string list of names</span></span>
<span><span class="va">mir.targets.score90</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">mir.targets.score90</span>,</span>
<span>                                         <span class="fu">stringr</span><span class="fu">::</span><span class="va"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span>,</span>
<span>                                         <span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="ecdf-for-log2foldchange-of-mirna-targets-compared-to-genes-with-seed-matches">ECDF for log2(FoldChange) of miRNA targets compared to genes with
seed matches<a class="anchor" aria-label="anchor" href="#ecdf-for-log2foldchange-of-mirna-targets-compared-to-genes-with-seed-matches"></a>
</h2>
<p>In this analysis we will compare genes with a <code>mer7m8</code>
seed match to those that are targets of miRNA expression in the DESeq2
experiments. We will also have a third group that compares the shared
gene targets of the siRNA and miRNA.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set of genes that are specific to mer7m8 seed matches</span></span>
<span><span class="va">mer7m8.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="op">!</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mir.targets.score90</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Set of genes that are specific to seed and miRNA</span></span>
<span><span class="va">seed_and_mir</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">res</span><span class="op">$</span><span class="va">mer7m8</span> <span class="op">&gt;=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mir.targets.score90</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Set of genes that is specific to miRNA only</span></span>
<span><span class="va">mir.targets</span> <span class="op">=</span> <span class="va">mir.targets.score90</span><span class="op">[</span><span class="op">!</span><span class="va">mir.targets.score90</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seed_and_mir</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Generate the background list of genes</span></span>
<span><span class="va">background.list</span> <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mer7m8.list</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mir.targets</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seed_and_mir</span><span class="op">)</span><span class="op">]</span></span>
<span>  </span>
<span><span class="va">ecdf.results</span> <span class="op">=</span> <span class="fu"><a href="../reference/deseq_fc_ecdf.html">deseq_fc_ecdf</a></span><span class="op">(</span><span class="va">res</span>, </span>
<span>                             title <span class="op">=</span> <span class="st">"Ttr D1 30mkg"</span>,</span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="va">background.list</span>, </span>
<span>                                  <span class="st">"mer7m8"</span> <span class="op">=</span> <span class="va">mer7m8.list</span>, </span>
<span>                                  <span class="st">"Top 10 miRNA"</span> <span class="op">=</span> <span class="va">mir.targets</span>, </span>
<span>                                  <span class="st">"Seed and miRNA"</span> <span class="op">=</span> <span class="va">seed_and_mir</span><span class="op">)</span>,</span>
<span>                             stats.test <span class="op">=</span> <span class="st">"KS"</span>, </span>
<span>                             factor.order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span>, </span>
<span>                                              <span class="st">"mer7m8"</span>, </span>
<span>                                              <span class="st">"Top 10 miRNA"</span>, </span>
<span>                                              <span class="st">"Seed and miRNA"</span><span class="op">)</span>, </span>
<span>                             null.name <span class="op">=</span> <span class="st">"Background"</span>,</span>
<span>                             target.name <span class="op">=</span> <span class="st">"Top 10 miRNA"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Comparing: Background vs. Top 10 miRNA"</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span></span>
<span><span class="va">ecdf.results</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="SeedMatchR_miRNA_analysis_files/figure-html/unnamed-chunk-14-1.png" width="960"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span> </span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] dplyr_1.1.2           org.Rn.eg.db_3.17.0   AnnotationDbi_1.62.1 </span></span>
<span><span class="co">#&gt;  [4] Biobase_2.60.0        rtracklayer_1.60.0    GenomicRanges_1.52.0 </span></span>
<span><span class="co">#&gt;  [7] GenomeInfoDb_1.36.0   IRanges_2.34.0        S4Vectors_0.38.1     </span></span>
<span><span class="co">#&gt; [10] BiocGenerics_0.46.0   SeedMatchR_0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.1.3                     bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [3] biomaRt_2.56.1                rlang_1.1.1                  </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.3                matrixStats_1.0.0            </span></span>
<span><span class="co">#&gt;   [7] compiler_4.3.1                RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;   [9] GenomicFeatures_1.52.0        png_0.1-8                    </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.0.4             vctrs_0.6.2                  </span></span>
<span><span class="co">#&gt;  [13] stringr_1.5.0                 pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.2                  fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [17] dbplyr_2.3.2                  XVector_0.40.0               </span></span>
<span><span class="co">#&gt;  [19] ellipsis_0.3.2                labeling_0.4.2               </span></span>
<span><span class="co">#&gt;  [21] utf8_1.2.3                    Rsamtools_2.16.0             </span></span>
<span><span class="co">#&gt;  [23] promises_1.2.0.1              rmarkdown_2.22               </span></span>
<span><span class="co">#&gt;  [25] ragg_1.2.5                    purrr_1.0.1                  </span></span>
<span><span class="co">#&gt;  [27] bit_4.0.5                     xfun_0.39                    </span></span>
<span><span class="co">#&gt;  [29] zlibbioc_1.46.0               cachem_1.0.8                 </span></span>
<span><span class="co">#&gt;  [31] jsonlite_1.8.5                progress_1.2.2               </span></span>
<span><span class="co">#&gt;  [33] blob_1.2.4                    highr_0.10                   </span></span>
<span><span class="co">#&gt;  [35] later_1.3.1                   DelayedArray_0.26.3          </span></span>
<span><span class="co">#&gt;  [37] BiocParallel_1.34.2           interactiveDisplayBase_1.38.0</span></span>
<span><span class="co">#&gt;  [39] parallel_4.3.1                prettyunits_1.1.1            </span></span>
<span><span class="co">#&gt;  [41] R6_2.5.1                      bslib_0.5.0                  </span></span>
<span><span class="co">#&gt;  [43] stringi_1.7.12                testit_0.13                  </span></span>
<span><span class="co">#&gt;  [45] jquerylib_0.1.4               Rcpp_1.0.10                  </span></span>
<span><span class="co">#&gt;  [47] SummarizedExperiment_1.30.2   knitr_1.43                   </span></span>
<span><span class="co">#&gt;  [49] httpuv_1.6.11                 Matrix_1.5-4.1               </span></span>
<span><span class="co">#&gt;  [51] tidyselect_1.2.0              yaml_2.3.7                   </span></span>
<span><span class="co">#&gt;  [53] codetools_0.2-19              curl_5.0.1                   </span></span>
<span><span class="co">#&gt;  [55] lattice_0.21-8                tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;  [57] shiny_1.7.4                   withr_2.5.0                  </span></span>
<span><span class="co">#&gt;  [59] KEGGREST_1.40.0               evaluate_0.21                </span></span>
<span><span class="co">#&gt;  [61] desc_1.4.2                    BiocFileCache_2.8.0          </span></span>
<span><span class="co">#&gt;  [63] xml2_1.3.4                    Biostrings_2.68.1            </span></span>
<span><span class="co">#&gt;  [65] pillar_1.9.0                  BiocManager_1.30.21          </span></span>
<span><span class="co">#&gt;  [67] filelock_1.0.2                MatrixGenerics_1.12.2        </span></span>
<span><span class="co">#&gt;  [69] generics_0.1.3                rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [71] RCurl_1.98-1.12               BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt;  [73] hms_1.1.3                     ggplot2_3.4.2                </span></span>
<span><span class="co">#&gt;  [75] munsell_0.5.0                 scales_1.2.1                 </span></span>
<span><span class="co">#&gt;  [77] xtable_1.8-4                  glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [79] tools_4.3.1                   AnnotationHub_3.8.0          </span></span>
<span><span class="co">#&gt;  [81] BiocIO_1.10.0                 GenomicAlignments_1.36.0     </span></span>
<span><span class="co">#&gt;  [83] fs_1.6.2                      XML_3.99-0.14                </span></span>
<span><span class="co">#&gt;  [85] cowplot_1.1.1                 grid_4.3.1                   </span></span>
<span><span class="co">#&gt;  [87] colorspace_2.1-0              GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt;  [89] restfulr_0.0.15               cli_3.6.1                    </span></span>
<span><span class="co">#&gt;  [91] rappdirs_0.3.3                textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [93] twosamples_2.0.0              fansi_1.0.4                  </span></span>
<span><span class="co">#&gt;  [95] S4Arrays_1.0.4                gtable_0.3.3                 </span></span>
<span><span class="co">#&gt;  [97] sass_0.4.6                    digest_0.6.31                </span></span>
<span><span class="co">#&gt;  [99] farver_2.1.1                  rjson_0.2.21                 </span></span>
<span><span class="co">#&gt; [101] memoise_2.0.1                 htmltools_0.5.5              </span></span>
<span><span class="co">#&gt; [103] pkgdown_2.0.7                 lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt; [105] httr_1.4.6                    mime_0.12                    </span></span>
<span><span class="co">#&gt; [107] bit64_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tareian Cazares.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
